%% BioMed_Central_Tex_Template_v1.06
%%                                      %
%  bmc_article.tex            ver: 1.06 %
%                                       %

%% For instructions on how to fill out this Tex template           %%
%% document please refer to Readme.html and the instructions for   %%
%% authors page on the biomed central website                      %%
%% http://www.biomedcentral.com/info/authors/                      %%

%%% additional documentclass options:
%  [doublespacing]
%  [linenumbers]   - put the line numbers on margins



%\documentclass[twocolumn]{bmcart}% uncomment this for twocolumn layout and comment line below
\documentclass[10pt,twocolumn,linenumbers]{article}

%%% Load packages
\usepackage{amsthm,amsmath}
\usepackage{graphicx}
\usepackage{booktabs}
\usepackage{caption}
\usepackage{float}
\RequirePackage{natbib}
\RequirePackage{hyperref}
\usepackage[utf8]{inputenc} %unicode support
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails

%\def\includegraphic{}
%\def\includegraphics{}

\title{Many-Sample de Novo Assembly and Analysis of Petromyzon marinus Transcriptome}

\author{
    Scott, Camille\\
    \texttt{camille.scott.w@gmail.com}   % email address
    \and
    Brown, C. Titus\\
    \texttt{ctbrown@ucdavis.edu}
}

%%% Begin ...
\begin{document}
\maketitle

\section*{Background}

Some biological background on sea lamprey, and why we care about them: ancestral jawed vertebrate, 
invasive species interest, regenerative capabilities, programmed genome rearrangement (PGR). This 
drives interest in a complete sea lamprey transcriptomic reference. [Cite: lamprey genome paper, 
more provided by weiming]

Background on NGS / RNAseq tech enabling deep mRNA sequencing. Lack of a complete reference due 
to PGR necessitates de novo assembly. De novo projects challenging because of difficulty in 
validation, data volume, data integrity. [Cite: ...]

Reproducibility crisis: many methods for pre-processing, assembly, and post-processing, but many 
difficult to replicate. Lack of documented tools, versions, parameters, source code for scripts. 
Demonstration of effective reproducible pipelines from start to finish. [Cite: assemblathon, mr-c, ...]

Goal: deeply characterize the sea lamprey transcriptome; produce a valuable resource for researchers; 
demonstrate efficacy of de novo approach for many-sample data; show open, reproducible pipelines.

\section*{Data Description}

Section should include information on sea lamprey sample prep where available, table of sample 
descriptions including read lengths, insert sizes, sequencing technology, tissue type, and conditions. 
Also includes accessions.

[Table 1, samples]

\section*{Results}

\subsection*{Many-sample de Novo Assembly}

Here we describe basic assembly statistics:

\begin{enumerate}

\item Number of transcripts, coding regions.
\item Number of orthologies, homologies (broken down by genome support).
\item Accuracy, completeness, contiguity (with discussion of limitations).

\end{enumerate}

\subsection*{de novo Assembly Improves Recall Over ab initio Predictions}

This subsection has two goals: show that our assembly is reasonably valid, and show that it has
better recall than lamp00. Though these two goals could ostensibly be split into two subsections, 
they have considerable overlap and mostly are shown by the same results. We make our case
through the following points:

\begin{enumerate}
\item Our assembly has good recall of existing genes in lamp00. We show that we cover lamp00 by
homology (blast lamp10 against lamp00), that we cover the gene products of lamp00 through 
annotation, and that we cover the GTF gene annotations on the genome.
\item Our assembly has good recall of core vertebrate orthologs. Here we present BUSCO results, 
both for lamp00 and lamp10.



\begin{table*}[t]
\caption {BUSCO Results}
\begin{center}

\begin{tabular}{lllllll}
\toprule
             &         & C(\%) & D(\%) & F(\%) & M(\%) &  n(\%) \\
\midrule
lamp10.fasta & metazoa &      &      &      &      &       \\
             & vertebrata &   66 &   43 &   27 &  5.9 &   843 \\
petMar2.cdna.fa & metazoa &   38 &   23 &   10 &   50 &  3023 \\
             & vertebrata &   48 &  6.4 &   15 &   36 &   843 \\
\bottomrule
\end{tabular}



\end{center}
\end{table*}

\item Our assembly extends existing genes. Can be extracted from homology with lamp00 and 
TransDecoder results.
\item Our assembly includes novel genes. We show this by filtering out transcripts with homology 
to the lamprey genome or transcriptome, and finding orthologies (recipricol best hits) for those 
that remain. We do another level of filtering for false positives with the mygene API.
\item Our assembly has already proven useful to lamprey researchers \citep{ren_genome-wide_2015}
(perhaps this goes in background?)
\end{enumerate}

Discussion point: do we split this into two sections?

% @ctb comments: consider dividing these results into 3 subsections:
% 1) Validation from genome
% 2) Validation from predicted transcripts (lamp00)
% 3) Validation from ortho/homologs


We find 73.93\%
 of annotations to be covered by a
transcript from lamp10. Breaking down this percentage by feature type reveals that the results are 
biased by the inclusion of gene and transcripts features, both of which tend to contain large stretches 
of intronic sequence unlikely to be covered above our chosen cutoff by any transcript. When we consider 
only exons, 80.71\%
 are covered, 
exons being a basic feature of mRNAs.

\begin{table*}[t]
\caption {Proportion of Annotations Covered by lamp00 and lamp10}
\begin{center}

\begin{tabular}{lrr}
\toprule
{} &    lamp00 &    lamp10 \\
\midrule
CDS         &  0.919196 &  0.814513 \\
UTR         &  0.957400 &  0.862251 \\
exon        &  0.896378 &  0.807120 \\
gene        &  0.051323 &  0.147640 \\
start\_codon &  0.960101 &  0.857505 \\
stop\_codon  &  0.636520 &  0.903922 \\
transcript  &  0.048511 &  0.138392 \\
\bottomrule
\end{tabular}



\end{center}
\end{table*}

Conversely, 29.72\%
 of transcripts are 
covered by a single feature, while 99.89\%
 of 
transcripts are covered in lamp00. We find the latter number encouraging; one would expect almost 
all transcripts in lamp00 to be covered by a single feature, as it was derived directly from the 
annotations, while previous evidence suggests that lamp10 is a superset of lamp00, thus explaining 
the disparity. Examining the extend to which our overlaps are a superset, we can break down transcript 
genome homologies by whether each transcript has only a homology, or both a homology and an annotation 
overlap, as follows.

\begin{table*}[H]
\caption {Proportion of Homologies and Annotation overlaps by Transcript in lamp00 and lamp10}
\begin{center}

\begin{tabular}{llrr}
\toprule
{} & assembly &     num &      prop \\
\midrule
presence    &          &         &           \\
+genome+ann &   lamp00 &   11476 &  0.998868 \\
+genome+ann &   lamp10 &  212606 &  0.297208 \\
+genome-ann &   lamp00 &       0 &  0.000000 \\
+genome-ann &   lamp10 &  311949 &  0.436082 \\
\bottomrule
\end{tabular}



\end{center}
\end{table*}

With so many transcripts having alignments to the genome but no corresponding annotation, it would
be valuable to further understand which of these transcripts have protein homologies to other
databases. In particular, lamprey's uniquely valuable position in vertebrate evolution drives 
questions regarding loss and gain of genes within gnathostomes. To that end, we have subdivided 
these transcripts based on their homologies and orthologies with both zebrafish and amphioxus.

% Add another column for not in genome at all

\begin{table*}[H]
\caption {BLAST Best Hits for Transcripts Filtered by Database Presence}
\begin{center}

\begin{tabular}{llrr}
\toprule
braflo\_best\_hom & danrer\_best\_hom &  no\_ann &  has\_ann \\
\midrule
           True &            True &   36605 &    10293 \\
           True &           False &     847 &      873 \\
          False &            True &    7196 &     4445 \\
          False &           False &  167958 &   296338 \\
\bottomrule
\end{tabular}



\end{center}
\end{table*}

\begin{table*}[H]
\caption {BLAST Orthologies for Transcripts Filtered by Database Presence}
\begin{center}

\begin{tabular}{llrr}
\toprule
danrer\_ortho & braflo\_ortho &  no\_ann &  has\_ann \\
\midrule
        True &         True &    3664 &      833 \\
        True &        False &    2450 &      892 \\
       False &         True &    1255 &      437 \\
       False &        False &  205237 &   309787 \\
\bottomrule
\end{tabular}



\end{center}
\end{table*}


Futher, \% of the genome is covered by annotations, while \% is covered by alignments from lamp10; 
\% of transcripts have any alignment to the genome. We also find that \% of transcript alignments 
entirely contain an annotation, increasing the annotation size by \%. \% of extensions are supported
by homology to a known protein. \% of transcript alignments are entirely contained by an annotation.



\subsection*{Improved recall discovers potential ancestral vertebrate genes}

Here we talk about the genes we have shown to potentially be ancestral vertebrate orthologs. This is
at least a useful result in its own right, but it would be nice to find something more compelling
here. Immune-related genes might be a good starting point. 

%The assembly was produced using the Trinity assembler \citep{haas_novo_2013} after pre-processing with an in-house pipeline, described further in the methods.

%Go over assembly metrics. [Cite: assessing denovo trans metrics]. Metrics to include are transcript count, percent reads mapping [Sup. Table, perc reads per sample], number ORFs, transcripts mapping to genome, transcripts annotated, transcripts mapping to genome but not annotated, average unigene OHR \citep{t_oneil_assessing_2013}, accuracy, contiguity, and completeness \citep{martin_next-generation_2011}.

%Discuss annotation: use of trinotate, blast databases, recipricol best-hits \citep{moreno-hagelsieb_choosing_2008}. 


\section*{Discussion}

\section*{Methods}


In order to assess the completeness of our de Novo transcriptome assembly (lamp10), we have compared
the alignment of the generated transcripts against the existing genome annotations released with
Pmarinus v7.0.75. First, we use blastn to align transcripts to the genome, using parameters 
`-evalue 1e-6`. Then, we use the coordinates from the annotation and the corresponding coordinates 
from the alignments to calculate the proportion of annotated sequence overlapped, proportion of
transcripts overlapped, and the respective proportions of non-overlapped sequence and transcripts.
We consider an annotated region to be overlapped by a trancript if it is at least 90\% covered, with
at least 98\% identity [TODO: get better justification for these cutoffs other than "things Camille 
remembers reading"].

We give particular attention to alignments which entirely contain annotated regions, as these
suggest extensions to existing annotations. When these alignments are from transcripts with homology
evidence from other species, we consider them to represent putative extensions [note: maybe not
necessary to establish validity, instead just report the numbers]. Further, alignments which are 
entirely contained within an annotation suggest either an overly aggressive prediction in the genome, 
or an incompletely assembled transcript.

\subsection*{Pre-processing}

Describe pipeline: Trimmomatic PE or SE; digital normalization to C=20 on each sample (PE and
orphans together for paired samples); pooled digital normalization C=20; filter-abund with variable
coverage C=2 Z=20 using table output from pooled digital normalization run.

\subsection*{Trinity Assembly}

Trinity assembly using all preprocessed reads. Final version will probably be with default settings.

\subsection*{Post-processing}

cd-hit-est (or vsearch) used to remove redundancy.
All transcripts aligned with BLASTX against zebrafish, amphioxus, mouse, lamprey, and human protein sequences
downloaded from ensembl, and with BLASTN against lamprey version 7.0.75 genome, CDS, mRNA, and
ncRNA. TransDecoder used to predict CDS, and hmmer used to make predictions against Pfam-A from
predicted proteins. bowtie2 used to align all raw reads against assembly, and eXpress used for
abundance estimation. Orthologies determined using recipricol best-hits (RBH). BUSCO ran to assess
recall of core vertebrate orthologs.

Orthologs were filtered by whether they had any blastn hit to lamprey resources; protein IDs then
queried with mygene to retreive gene symbols associated with each transcript, and symbols queried
using the taxonomy tree option to determine gene membership in gnathostomata, cylcostomata, and
cephalochordata lineages.


% if your bibliography is in bibtex format, use those commands:
%\bibliographystyle{include/bmc-mathphys} % Style BST file
\bibliography{2015-petMarSB}{}      % Bibliography file (usually '*.bib' )
\bibliographystyle{plain}

% or include bibliography directly:
% \begin{thebibliography}
% \bibitem{b1}
% \end{thebibliography}

\end{document}
