{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "# Moore Foundation Site Visit\n",
    "## Camille Scott: \"What is it I do, anyway?\"\n",
    "### 2015-09-15\n",
    "### UC Davis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true,
    "slideshow": {
     "slide_type": "skip"
    }
   },
   "outputs": [],
   "source": [
    "from IPython.display import HTML"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "* Data processing logistics are crucial to understanding results in bioinformatics; they are rarely adequately reported. Pydoit!\n",
    "  - http://pydoit.org/\n",
    "  - https://github.com/dib-lab/2015-petMarSB <-- This is public!\n",
    "  - https://github.com/dib-lab/2015-petMarSB/blob/master/notebooks/petmar-genome-completeness.ipynb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false,
    "scrolled": true,
    "slideshow": {
     "slide_type": "subslide"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "\n",
       "\n",
       "<!DOCTYPE html>\n",
       "<html lang=\"en\" class=\"\">\n",
       "  <head prefix=\"og: http://ogp.me/ns# fb: http://ogp.me/ns/fb# object: http://ogp.me/ns/object# article: http://ogp.me/ns/article# profile: http://ogp.me/ns/profile#\">\n",
       "    <meta charset='utf-8'>\n",
       "    <meta http-equiv=\"X-UA-Compatible\" content=\"IE=edge\">\n",
       "    <meta http-equiv=\"Content-Language\" content=\"en\">\n",
       "    <meta name=\"viewport\" content=\"width=1020\">\n",
       "    \n",
       "    \n",
       "    <title>dib-lab/2015-petMarSB Â· GitHub</title>\n",
       "    <link rel=\"search\" type=\"application/opensearchdescription+xml\" href=\"/opensearch.xml\" title=\"GitHub\">\n",
       "    <link rel=\"fluid-icon\" href=\"https://github.com/fluidicon.png\" title=\"GitHub\">\n",
       "    <link rel=\"apple-touch-icon\" sizes=\"57x57\" href=\"/apple-touch-icon-114.png\">\n",
       "    <link rel=\"apple-touch-icon\" sizes=\"114x114\" href=\"/apple-touch-icon-114.png\">\n",
       "    <link rel=\"apple-touch-icon\" sizes=\"72x72\" href=\"/apple-touch-icon-144.png\">\n",
       "    <link rel=\"apple-touch-icon\" sizes=\"144x144\" href=\"/apple-touch-icon-144.png\">\n",
       "    <meta property=\"fb:app_id\" content=\"1401488693436528\">\n",
       "\n",
       "      <meta content=\"@github\" name=\"twitter:site\" /><meta content=\"summary\" name=\"twitter:card\" /><meta content=\"dib-lab/2015-petMarSB\" name=\"twitter:title\" /><meta content=\"2015-petMarSB - Many-sample de novo assembly of the sea lamprey transcriptome\" name=\"twitter:description\" /><meta content=\"https://avatars1.githubusercontent.com/u/1740697?v=3&amp;s=400\" name=\"twitter:image:src\" />\n",
       "      <meta content=\"GitHub\" property=\"og:site_name\" /><meta content=\"object\" property=\"og:type\" /><meta content=\"https://avatars1.githubusercontent.com/u/1740697?v=3&amp;s=400\" property=\"og:image\" /><meta content=\"dib-lab/2015-petMarSB\" property=\"og:title\" /><meta content=\"https://github.com/dib-lab/2015-petMarSB\" property=\"og:url\" /><meta content=\"2015-petMarSB - Many-sample de novo assembly of the sea lamprey transcriptome\" property=\"og:description\" />\n",
       "      <meta name=\"browser-stats-url\" content=\"https://api.github.com/_private/browser/stats\">\n",
       "    <meta name=\"browser-errors-url\" content=\"https://api.github.com/_private/browser/errors\">\n",
       "    <link rel=\"assets\" href=\"https://assets-cdn.github.com/\">\n",
       "    \n",
       "    <meta name=\"pjax-timeout\" content=\"1000\">\n",
       "    \n",
       "\n",
       "    <meta name=\"msapplication-TileImage\" content=\"/windows-tile.png\">\n",
       "    <meta name=\"msapplication-TileColor\" content=\"#ffffff\">\n",
       "    <meta name=\"selected-link\" value=\"repo_source\" data-pjax-transient>\n",
       "\n",
       "    <meta name=\"google-site-verification\" content=\"KT5gs8h0wvaagLKAVWq8bbeNwnZZK1r1XQysX3xurLU\">\n",
       "        <meta name=\"google-analytics\" content=\"UA-3769691-2\">\n",
       "\n",
       "    <meta content=\"collector.githubapp.com\" name=\"octolytics-host\" /><meta content=\"collector-cdn.github.com\" name=\"octolytics-script-host\" /><meta content=\"github\" name=\"octolytics-app-id\" /><meta content=\"A9EDADC9:75C8:860BBF:55F86095\" name=\"octolytics-dimension-request_id\" />\n",
       "    \n",
       "    <meta content=\"Rails, view, files#disambiguate\" data-pjax-transient=\"true\" name=\"analytics-event\" />\n",
       "    <meta class=\"js-ga-set\" name=\"dimension1\" content=\"Logged Out\">\n",
       "      <meta class=\"js-ga-set\" name=\"dimension4\" content=\"Current repo nav\">\n",
       "    <meta name=\"is-dotcom\" content=\"true\">\n",
       "        <meta name=\"hostname\" content=\"github.com\">\n",
       "    <meta name=\"user-login\" content=\"\">\n",
       "\n",
       "      <link rel=\"mask-icon\" href=\"https://assets-cdn.github.com/pinned-octocat.svg\" color=\"#4078c0\">\n",
       "      <link rel=\"icon\" type=\"image/x-icon\" href=\"https://assets-cdn.github.com/favicon.ico\">\n",
       "\n",
       "    <!-- </textarea> --><!-- '\"` --><meta content=\"authenticity_token\" name=\"csrf-param\" />\n",
       "<meta content=\"lSfEercRqQJUF0M90Pf0cbwI4vnQBySfHUmm1fG3dV/ZMqTKws4uXNRrmts/iNPIpQx6byDGU18m8OhQpkB2Sg==\" name=\"csrf-token\" />\n",
       "    \n",
       "\n",
       "    <link crossorigin=\"anonymous\" href=\"https://assets-cdn.github.com/assets/github-d1b5abdc908e0d7b1177dec08377119b557ebf6285955d1d4b3a49990b7f2f95.css\" media=\"all\" rel=\"stylesheet\" />\n",
       "    <link crossorigin=\"anonymous\" href=\"https://assets-cdn.github.com/assets/github2-29104553f446a6d7b0da67330b1c4a5b45b3f0b04e24c29abbc37f6695c27e4b.css\" media=\"all\" rel=\"stylesheet\" />\n",
       "    \n",
       "    \n",
       "\n",
       "\n",
       "    <meta http-equiv=\"x-pjax-version\" content=\"51aa40ccaf5953beea475a104cb2b86c\">\n",
       "\n",
       "      \n",
       "  <meta name=\"description\" content=\"2015-petMarSB - Many-sample de novo assembly of the sea lamprey transcriptome\">\n",
       "  <meta name=\"go-import\" content=\"github.com/dib-lab/2015-petMarSB git https://github.com/dib-lab/2015-petMarSB.git\">\n",
       "\n",
       "  <meta content=\"1740697\" name=\"octolytics-dimension-user_id\" /><meta content=\"dib-lab\" name=\"octolytics-dimension-user_login\" /><meta content=\"38009438\" name=\"octolytics-dimension-repository_id\" /><meta content=\"dib-lab/2015-petMarSB\" name=\"octolytics-dimension-repository_nwo\" /><meta content=\"true\" name=\"octolytics-dimension-repository_public\" /><meta content=\"true\" name=\"octolytics-dimension-repository_is_fork\" /><meta content=\"33301748\" name=\"octolytics-dimension-repository_parent_id\" /><meta content=\"camillescott/2015-petMarSB\" name=\"octolytics-dimension-repository_parent_nwo\" /><meta content=\"33301748\" name=\"octolytics-dimension-repository_network_root_id\" /><meta content=\"camillescott/2015-petMarSB\" name=\"octolytics-dimension-repository_network_root_nwo\" />\n",
       "  <link href=\"https://github.com/dib-lab/2015-petMarSB/commits/master.atom\" rel=\"alternate\" title=\"Recent Commits to 2015-petMarSB:master\" type=\"application/atom+xml\">\n",
       "\n",
       "  </head>\n",
       "\n",
       "\n",
       "  <body class=\"logged_out  env-production  vis-public fork\">\n",
       "    <a href=\"#start-of-content\" tabindex=\"1\" class=\"accessibility-aid js-skip-to-content\">Skip to content</a>\n",
       "\n",
       "    \n",
       "    \n",
       "    \n",
       "\n",
       "\n",
       "\n",
       "      \n",
       "      <div class=\"header header-logged-out\" role=\"banner\">\n",
       "  <div class=\"container clearfix\">\n",
       "\n",
       "    <a class=\"header-logo-wordmark\" href=\"https://github.com/\" data-ga-click=\"(Logged out) Header, go to homepage, icon:logo-wordmark\">\n",
       "      <span class=\"mega-octicon octicon-logo-github\"></span>\n",
       "    </a>\n",
       "\n",
       "    <div class=\"header-actions\" role=\"navigation\">\n",
       "        <a class=\"btn btn-primary\" href=\"/join\" data-ga-click=\"(Logged out) Header, clicked Sign up, text:sign-up\">Sign up</a>\n",
       "      <a class=\"btn\" href=\"/login?return_to=%2Fdib-lab%2F2015-petMarSB\" data-ga-click=\"(Logged out) Header, clicked Sign in, text:sign-in\">Sign in</a>\n",
       "    </div>\n",
       "\n",
       "    <div class=\"site-search repo-scope js-site-search\" role=\"search\">\n",
       "      <!-- </textarea> --><!-- '\"` --><form accept-charset=\"UTF-8\" action=\"/dib-lab/2015-petMarSB/search\" class=\"js-site-search-form\" data-global-search-url=\"/search\" data-repo-search-url=\"/dib-lab/2015-petMarSB/search\" method=\"get\"><div style=\"margin:0;padding:0;display:inline\"><input name=\"utf8\" type=\"hidden\" value=\"&#x2713;\" /></div>\n",
       "  <label class=\"js-chromeless-input-container form-control\">\n",
       "    <div class=\"scope-badge\">This repository</div>\n",
       "    <input type=\"text\"\n",
       "      class=\"js-site-search-focus js-site-search-field is-clearable chromeless-input\"\n",
       "      data-hotkey=\"s\"\n",
       "      name=\"q\"\n",
       "      placeholder=\"Search\"\n",
       "      aria-label=\"Search this repository\"\n",
       "      data-global-scope-placeholder=\"Search GitHub\"\n",
       "      data-repo-scope-placeholder=\"Search\"\n",
       "      tabindex=\"1\"\n",
       "      autocapitalize=\"off\">\n",
       "  </label>\n",
       "</form>\n",
       "    </div>\n",
       "\n",
       "      <ul class=\"header-nav left\" role=\"navigation\">\n",
       "          <li class=\"header-nav-item\">\n",
       "            <a class=\"header-nav-link\" href=\"/explore\" data-ga-click=\"(Logged out) Header, go to explore, text:explore\">Explore</a>\n",
       "          </li>\n",
       "          <li class=\"header-nav-item\">\n",
       "            <a class=\"header-nav-link\" href=\"/features\" data-ga-click=\"(Logged out) Header, go to features, text:features\">Features</a>\n",
       "          </li>\n",
       "          <li class=\"header-nav-item\">\n",
       "            <a class=\"header-nav-link\" href=\"https://enterprise.github.com/\" data-ga-click=\"(Logged out) Header, go to enterprise, text:enterprise\">Enterprise</a>\n",
       "          </li>\n",
       "          <li class=\"header-nav-item\">\n",
       "            <a class=\"header-nav-link\" href=\"/pricing\" data-ga-click=\"(Logged out) Header, go to pricing, text:pricing\">Pricing</a>\n",
       "          </li>\n",
       "      </ul>\n",
       "\n",
       "  </div>\n",
       "</div>\n",
       "\n",
       "\n",
       "\n",
       "    <div id=\"start-of-content\" class=\"accessibility-aid\"></div>\n",
       "\n",
       "    <div id=\"js-flash-container\">\n",
       "</div>\n",
       "\n",
       "\n",
       "    <div role=\"main\" class=\"main-content\">\n",
       "        <div itemscope itemtype=\"http://schema.org/WebPage\">\n",
       "    <div class=\"pagehead repohead instapaper_ignore readability-menu\">\n",
       "      <div class=\"container\">\n",
       "\n",
       "        <div class=\"clearfix\">\n",
       "          \n",
       "<ul class=\"pagehead-actions\">\n",
       "\n",
       "  <li>\n",
       "      <a href=\"/login?return_to=%2Fdib-lab%2F2015-petMarSB\"\n",
       "    class=\"btn btn-sm btn-with-count tooltipped tooltipped-n\"\n",
       "    aria-label=\"You must be signed in to watch a repository\" rel=\"nofollow\">\n",
       "    <span class=\"octicon octicon-eye\"></span>\n",
       "    Watch\n",
       "  </a>\n",
       "  <a class=\"social-count\" href=\"/dib-lab/2015-petMarSB/watchers\">\n",
       "    2\n",
       "  </a>\n",
       "\n",
       "  </li>\n",
       "\n",
       "  <li>\n",
       "      <a href=\"/login?return_to=%2Fdib-lab%2F2015-petMarSB\"\n",
       "    class=\"btn btn-sm btn-with-count tooltipped tooltipped-n\"\n",
       "    aria-label=\"You must be signed in to star a repository\" rel=\"nofollow\">\n",
       "    <span class=\"octicon octicon-star\"></span>\n",
       "    Star\n",
       "  </a>\n",
       "\n",
       "    <a class=\"social-count js-social-count\" href=\"/dib-lab/2015-petMarSB/stargazers\">\n",
       "      0\n",
       "    </a>\n",
       "\n",
       "  </li>\n",
       "\n",
       "    <li>\n",
       "      <a href=\"/login?return_to=%2Fdib-lab%2F2015-petMarSB\"\n",
       "        class=\"btn btn-sm btn-with-count tooltipped tooltipped-n\"\n",
       "        aria-label=\"You must be signed in to fork a repository\" rel=\"nofollow\">\n",
       "        <span class=\"octicon octicon-repo-forked\"></span>\n",
       "        Fork\n",
       "      </a>\n",
       "      <a href=\"/dib-lab/2015-petMarSB/network\" class=\"social-count\">\n",
       "        1\n",
       "      </a>\n",
       "    </li>\n",
       "</ul>\n",
       "\n",
       "          <h1 itemscope itemtype=\"http://data-vocabulary.org/Breadcrumb\" class=\"entry-title public \">\n",
       "  <span class=\"mega-octicon octicon-repo-forked\"></span>\n",
       "  <span class=\"author\"><a href=\"/dib-lab\" class=\"url fn\" itemprop=\"url\" rel=\"author\"><span itemprop=\"title\">dib-lab</span></a></span><!--\n",
       "--><span class=\"path-divider\">/</span><!--\n",
       "--><strong><a href=\"/dib-lab/2015-petMarSB\" data-pjax=\"#js-repo-pjax-container\">2015-petMarSB</a></strong>\n",
       "\n",
       "  <span class=\"page-context-loader\">\n",
       "    <img alt=\"\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "  </span>\n",
       "\n",
       "    <span class=\"fork-flag\">\n",
       "      <span class=\"text\">forked from <a href=\"/camillescott/2015-petMarSB\">camillescott/2015-petMarSB</a></span>\n",
       "    </span>\n",
       "</h1>\n",
       "\n",
       "        </div>\n",
       "      </div>\n",
       "    </div>\n",
       "\n",
       "    <div class=\"container\">\n",
       "      <div class=\"repository-with-sidebar repo-container new-discussion-timeline with-full-navigation\">\n",
       "        <div class=\"repository-sidebar clearfix\">\n",
       "          \n",
       "<nav class=\"sunken-menu repo-nav js-repo-nav js-sidenav-container-pjax js-octicon-loaders\"\n",
       "     role=\"navigation\"\n",
       "     data-pjax=\"#js-repo-pjax-container\"\n",
       "     data-issue-count-url=\"/dib-lab/2015-petMarSB/issues/counts\">\n",
       "  <ul class=\"sunken-menu-group\">\n",
       "    <li class=\"tooltipped tooltipped-w\" aria-label=\"Code\">\n",
       "      <a href=\"/dib-lab/2015-petMarSB\" aria-label=\"Code\" aria-selected=\"true\" class=\"js-selected-navigation-item selected sunken-menu-item\" data-hotkey=\"g c\" data-selected-links=\"repo_source repo_downloads repo_commits repo_releases repo_tags repo_branches /dib-lab/2015-petMarSB\">\n",
       "        <span class=\"octicon octicon-code\"></span> <span class=\"full-word\">Code</span>\n",
       "        <img alt=\"\" class=\"mini-loader\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "</a>    </li>\n",
       "\n",
       "      <li class=\"tooltipped tooltipped-w\" aria-label=\"Issues\">\n",
       "        <a href=\"/dib-lab/2015-petMarSB/issues\" aria-label=\"Issues\" class=\"js-selected-navigation-item sunken-menu-item\" data-hotkey=\"g i\" data-selected-links=\"repo_issues repo_labels repo_milestones /dib-lab/2015-petMarSB/issues\">\n",
       "          <span class=\"octicon octicon-issue-opened\"></span> <span class=\"full-word\">Issues</span>\n",
       "          <span class=\"js-issue-replace-counter\"></span>\n",
       "          <img alt=\"\" class=\"mini-loader\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "</a>      </li>\n",
       "\n",
       "    <li class=\"tooltipped tooltipped-w\" aria-label=\"Pull requests\">\n",
       "      <a href=\"/dib-lab/2015-petMarSB/pulls\" aria-label=\"Pull requests\" class=\"js-selected-navigation-item sunken-menu-item\" data-hotkey=\"g p\" data-selected-links=\"repo_pulls /dib-lab/2015-petMarSB/pulls\">\n",
       "          <span class=\"octicon octicon-git-pull-request\"></span> <span class=\"full-word\">Pull requests</span>\n",
       "          <span class=\"js-pull-replace-counter\"></span>\n",
       "          <img alt=\"\" class=\"mini-loader\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "</a>    </li>\n",
       "\n",
       "      <li class=\"tooltipped tooltipped-w\" aria-label=\"Wiki\">\n",
       "        <a href=\"/dib-lab/2015-petMarSB/wiki\" aria-label=\"Wiki\" class=\"js-selected-navigation-item sunken-menu-item\" data-hotkey=\"g w\" data-selected-links=\"repo_wiki /dib-lab/2015-petMarSB/wiki\">\n",
       "          <span class=\"octicon octicon-book\"></span> <span class=\"full-word\">Wiki</span>\n",
       "          <img alt=\"\" class=\"mini-loader\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "</a>      </li>\n",
       "  </ul>\n",
       "  <div class=\"sunken-menu-separator\"></div>\n",
       "  <ul class=\"sunken-menu-group\">\n",
       "\n",
       "    <li class=\"tooltipped tooltipped-w\" aria-label=\"Pulse\">\n",
       "      <a href=\"/dib-lab/2015-petMarSB/pulse\" aria-label=\"Pulse\" class=\"js-selected-navigation-item sunken-menu-item\" data-selected-links=\"pulse /dib-lab/2015-petMarSB/pulse\">\n",
       "        <span class=\"octicon octicon-pulse\"></span> <span class=\"full-word\">Pulse</span>\n",
       "        <img alt=\"\" class=\"mini-loader\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "</a>    </li>\n",
       "\n",
       "    <li class=\"tooltipped tooltipped-w\" aria-label=\"Graphs\">\n",
       "      <a href=\"/dib-lab/2015-petMarSB/graphs\" aria-label=\"Graphs\" class=\"js-selected-navigation-item sunken-menu-item\" data-selected-links=\"repo_graphs repo_contributors /dib-lab/2015-petMarSB/graphs\">\n",
       "        <span class=\"octicon octicon-graph\"></span> <span class=\"full-word\">Graphs</span>\n",
       "        <img alt=\"\" class=\"mini-loader\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "</a>    </li>\n",
       "  </ul>\n",
       "\n",
       "\n",
       "</nav>\n",
       "\n",
       "            <div class=\"only-with-full-nav\">\n",
       "                \n",
       "<div class=\"js-clone-url clone-url open\"\n",
       "  data-protocol-type=\"http\">\n",
       "  <h3><span class=\"text-emphasized\">HTTPS</span> clone URL</h3>\n",
       "  <div class=\"input-group js-zeroclipboard-container\">\n",
       "    <input type=\"text\" class=\"input-mini input-monospace js-url-field js-zeroclipboard-target\"\n",
       "           value=\"https://github.com/dib-lab/2015-petMarSB.git\" readonly=\"readonly\" aria-label=\"HTTPS clone URL\">\n",
       "    <span class=\"input-group-button\">\n",
       "      <button aria-label=\"Copy to clipboard\" class=\"js-zeroclipboard btn btn-sm zeroclipboard-button tooltipped tooltipped-s\" data-copied-hint=\"Copied!\" type=\"button\"><span class=\"octicon octicon-clippy\"></span></button>\n",
       "    </span>\n",
       "  </div>\n",
       "</div>\n",
       "\n",
       "  \n",
       "<div class=\"js-clone-url clone-url \"\n",
       "  data-protocol-type=\"subversion\">\n",
       "  <h3><span class=\"text-emphasized\">Subversion</span> checkout URL</h3>\n",
       "  <div class=\"input-group js-zeroclipboard-container\">\n",
       "    <input type=\"text\" class=\"input-mini input-monospace js-url-field js-zeroclipboard-target\"\n",
       "           value=\"https://github.com/dib-lab/2015-petMarSB\" readonly=\"readonly\" aria-label=\"Subversion checkout URL\">\n",
       "    <span class=\"input-group-button\">\n",
       "      <button aria-label=\"Copy to clipboard\" class=\"js-zeroclipboard btn btn-sm zeroclipboard-button tooltipped tooltipped-s\" data-copied-hint=\"Copied!\" type=\"button\"><span class=\"octicon octicon-clippy\"></span></button>\n",
       "    </span>\n",
       "  </div>\n",
       "</div>\n",
       "\n",
       "\n",
       "\n",
       "  <div class=\"clone-options\">You can clone with\n",
       "    <!-- </textarea> --><!-- '\"` --><form accept-charset=\"UTF-8\" action=\"/users/set_protocol?protocol_selector=http&amp;protocol_type=clone\" class=\"inline-form js-clone-selector-form \" data-form-nonce=\"df911216e7615289d802116f1dba482365942e8a\" data-remote=\"true\" method=\"post\"><div style=\"margin:0;padding:0;display:inline\"><input name=\"utf8\" type=\"hidden\" value=\"&#x2713;\" /><input name=\"authenticity_token\" type=\"hidden\" value=\"PVpLPuSTo07zzXaZxKGbP4LNyWJybOMQ7hcrub769KUbvZe5d8lk9YDd+bX29qpgGnS0M38bqGomNpl6lB1hAQ==\" /></div><button class=\"btn-link js-clone-selector\" data-protocol=\"http\" type=\"submit\">HTTPS</button></form> or <!-- </textarea> --><!-- '\"` --><form accept-charset=\"UTF-8\" action=\"/users/set_protocol?protocol_selector=subversion&amp;protocol_type=clone\" class=\"inline-form js-clone-selector-form \" data-form-nonce=\"df911216e7615289d802116f1dba482365942e8a\" data-remote=\"true\" method=\"post\"><div style=\"margin:0;padding:0;display:inline\"><input name=\"utf8\" type=\"hidden\" value=\"&#x2713;\" /><input name=\"authenticity_token\" type=\"hidden\" value=\"B1JuleFv1dTEhDPxdti6HLM0JjwmFcAoGyPR46pp+8r4i/9beVGDQ6QaLdtA0XBuuu5bq3p/1oNmz4uYRNeJWQ==\" /></div><button class=\"btn-link js-clone-selector\" data-protocol=\"subversion\" type=\"submit\">Subversion</button></form>.\n",
       "    <a href=\"https://help.github.com/articles/which-remote-url-should-i-use\" class=\"help tooltipped tooltipped-n\" aria-label=\"Get help on which URL is right for you.\">\n",
       "      <span class=\"octicon octicon-question\"></span>\n",
       "    </a>\n",
       "  </div>\n",
       "\n",
       "              <a href=\"/dib-lab/2015-petMarSB/archive/master.zip\"\n",
       "                 class=\"btn btn-sm sidebar-button\"\n",
       "                 aria-label=\"Download the contents of dib-lab/2015-petMarSB as a zip file\"\n",
       "                 title=\"Download the contents of dib-lab/2015-petMarSB as a zip file\"\n",
       "                 rel=\"nofollow\">\n",
       "                <span class=\"octicon octicon-cloud-download\"></span>\n",
       "                Download ZIP\n",
       "              </a>\n",
       "            </div>\n",
       "        </div>\n",
       "        <div id=\"js-repo-pjax-container\" class=\"repository-content context-loader-container\" data-pjax-container>\n",
       "\n",
       "          \n",
       "<span id=\"js-show-full-navigation\"></span>\n",
       "\n",
       "  <div class=\"repository-meta js-details-container \">\n",
       "    <div class=\"repository-description\">\n",
       "      Many-sample de novo assembly of the sea lamprey transcriptome\n",
       "    </div>\n",
       "\n",
       "\n",
       "</div>\n",
       "\n",
       "\n",
       "<div class=\"overall-summary overall-summary-bottomless\">\n",
       "\n",
       "  <div class=\"stats-switcher-viewport js-stats-switcher-viewport\">\n",
       "    <div class=\"stats-switcher-wrapper\">\n",
       "    <ul class=\"numbers-summary\">\n",
       "      <li class=\"commits\">\n",
       "        <a data-pjax href=\"/dib-lab/2015-petMarSB/commits/master\">\n",
       "            <span class=\"octicon octicon-history\"></span>\n",
       "            <span class=\"num text-emphasized\">\n",
       "              174\n",
       "            </span>\n",
       "            commits\n",
       "        </a>\n",
       "      </li>\n",
       "      <li>\n",
       "        <a data-pjax href=\"/dib-lab/2015-petMarSB/branches\">\n",
       "          <span class=\"octicon octicon-git-branch\"></span>\n",
       "          <span class=\"num text-emphasized\">\n",
       "            2\n",
       "          </span>\n",
       "          branches\n",
       "        </a>\n",
       "      </li>\n",
       "\n",
       "      <li>\n",
       "        <a data-pjax href=\"/dib-lab/2015-petMarSB/releases\">\n",
       "          <span class=\"octicon octicon-tag\"></span>\n",
       "          <span class=\"num text-emphasized\">\n",
       "            0\n",
       "          </span>\n",
       "          releases\n",
       "        </a>\n",
       "      </li>\n",
       "\n",
       "      <li>\n",
       "        <include-fragment src=\"/dib-lab/2015-petMarSB/contributors_size\">\n",
       "          <a href=\"/dib-lab/2015-petMarSB/graphs/contributors\">\n",
       "            <span class=\"octicon octicon-organization\"></span>\n",
       "            <span class=\"num text-emphasized\"></span>\n",
       "            Fetching contributors\n",
       "          </a>\n",
       "</include-fragment>      </li>\n",
       "    </ul>\n",
       "\n",
       "      <div class=\"repository-lang-stats\">\n",
       "        <ol class=\"repository-lang-stats-numbers\">\n",
       "          <li>\n",
       "              <a href=\"/dib-lab/2015-petMarSB/search?l=python\">\n",
       "                <span class=\"color-block language-color\" style=\"background-color:#3572A5;\"></span>\n",
       "                <span class=\"lang\">Python</span>\n",
       "                <span class=\"percent\">97.4%</span>\n",
       "              </a>\n",
       "          </li>\n",
       "          <li>\n",
       "              <a href=\"/dib-lab/2015-petMarSB/search?l=bash\">\n",
       "                <span class=\"color-block language-color\" style=\"background-color:#89e051;\"></span>\n",
       "                <span class=\"lang\">Shell</span>\n",
       "                <span class=\"percent\">1.6%</span>\n",
       "              </a>\n",
       "          </li>\n",
       "          <li>\n",
       "              <a href=\"/dib-lab/2015-petMarSB/search?l=makefile\">\n",
       "                <span class=\"color-block language-color\" style=\"background-color:#427819;\"></span>\n",
       "                <span class=\"lang\">Makefile</span>\n",
       "                <span class=\"percent\">1.0%</span>\n",
       "              </a>\n",
       "          </li>\n",
       "        </ol>\n",
       "      </div>\n",
       "    </div>\n",
       "  </div>\n",
       "\n",
       "</div>\n",
       "\n",
       "  <div class=\"repository-lang-stats-graph js-toggle-lang-stats\" title=\"Click for language details\">\n",
       "    <span class=\"language-color\" aria-label=\"Python 97.4%\" style=\"width:97.4%; background-color:#3572A5;\" itemprop=\"keywords\">Python</span>\n",
       "    <span class=\"language-color\" aria-label=\"Shell 1.6%\" style=\"width:1.6%; background-color:#89e051;\" itemprop=\"keywords\">Shell</span>\n",
       "    <span class=\"language-color\" aria-label=\"Makefile 1.0%\" style=\"width:1.0%; background-color:#427819;\" itemprop=\"keywords\">Makefile</span>\n",
       "  </div>\n",
       "\n",
       "<include-fragment src=\"/dib-lab/2015-petMarSB/show_partial?partial=tree%2Frecently_touched_branches_list\"></include-fragment>\n",
       "\n",
       "  <div class=\"file-navigation in-mid-page\">\n",
       "  <a href=\"/dib-lab/2015-petMarSB/find/master\"\n",
       "        class=\"js-show-file-finder btn btn-sm empty-icon tooltipped tooltipped-s right\"\n",
       "        data-pjax\n",
       "        data-hotkey=\"t\"\n",
       "        aria-label=\"Quickly jump between files\"\n",
       "        data-ga-click=\"Repository, find file, location:repo overview\">\n",
       "    <span class=\"octicon octicon-list-unordered\"></span>\n",
       "  </a>\n",
       "    <a href=\"/dib-lab/2015-petMarSB/compare\" aria-label=\"Compare, review, create a pull request\" class=\"btn btn-sm btn-primary tooltipped tooltipped-se left compare-button\" aria-label=\"Compare &amp; review\" data-pjax data-ga-click=\"Repository, go to compare view, location:repo overview; icon:git-compare\">\n",
       "      <span class=\"octicon octicon-git-compare\"></span>\n",
       "    </a>\n",
       "\n",
       "  \n",
       "<div class=\"select-menu js-menu-container js-select-menu left\">\n",
       "  <span class=\"btn btn-sm select-menu-button js-menu-target css-truncate\" data-hotkey=\"w\"\n",
       "    data-ref=\"master\"\n",
       "    title=\"master\"\n",
       "    role=\"button\" aria-label=\"Switch branches or tags\" tabindex=\"0\" aria-haspopup=\"true\">\n",
       "    <i>Branch:</i>\n",
       "    <span class=\"js-select-button css-truncate-target\">master</span>\n",
       "  </span>\n",
       "\n",
       "  <div class=\"select-menu-modal-holder js-menu-content js-navigation-container\" data-pjax aria-hidden=\"true\">\n",
       "\n",
       "    <div class=\"select-menu-modal\">\n",
       "      <div class=\"select-menu-header\">\n",
       "        <span class=\"select-menu-title\">Switch branches/tags</span>\n",
       "        <span class=\"octicon octicon-x js-menu-close\" role=\"button\" aria-label=\"Close\"></span>\n",
       "      </div>\n",
       "\n",
       "      <div class=\"select-menu-filters\">\n",
       "        <div class=\"select-menu-text-filter\">\n",
       "          <input type=\"text\" aria-label=\"Filter branches/tags\" id=\"context-commitish-filter-field\" class=\"js-filterable-field js-navigation-enable\" placeholder=\"Filter branches/tags\">\n",
       "        </div>\n",
       "        <div class=\"select-menu-tabs\">\n",
       "          <ul>\n",
       "            <li class=\"select-menu-tab\">\n",
       "              <a href=\"#\" data-tab-filter=\"branches\" data-filter-placeholder=\"Filter branches/tags\" class=\"js-select-menu-tab\" role=\"tab\">Branches</a>\n",
       "            </li>\n",
       "            <li class=\"select-menu-tab\">\n",
       "              <a href=\"#\" data-tab-filter=\"tags\" data-filter-placeholder=\"Find a tagâ¦\" class=\"js-select-menu-tab\" role=\"tab\">Tags</a>\n",
       "            </li>\n",
       "          </ul>\n",
       "        </div>\n",
       "      </div>\n",
       "\n",
       "      <div class=\"select-menu-list select-menu-tab-bucket js-select-menu-tab-bucket\" data-tab-filter=\"branches\" role=\"menu\">\n",
       "\n",
       "        <div data-filterable-for=\"context-commitish-filter-field\" data-filterable-type=\"substring\">\n",
       "\n",
       "\n",
       "            <a class=\"select-menu-item js-navigation-item js-navigation-open \"\n",
       "               href=\"/dib-lab/2015-petMarSB/tree/make_package\"\n",
       "               data-name=\"make_package\"\n",
       "               data-skip-pjax=\"true\"\n",
       "               rel=\"nofollow\">\n",
       "              <span class=\"select-menu-item-icon octicon octicon-check\"></span>\n",
       "              <span class=\"select-menu-item-text css-truncate-target\" title=\"make_package\">\n",
       "                make_package\n",
       "              </span>\n",
       "            </a>\n",
       "            <a class=\"select-menu-item js-navigation-item js-navigation-open selected\"\n",
       "               href=\"/dib-lab/2015-petMarSB/tree/master\"\n",
       "               data-name=\"master\"\n",
       "               data-skip-pjax=\"true\"\n",
       "               rel=\"nofollow\">\n",
       "              <span class=\"select-menu-item-icon octicon octicon-check\"></span>\n",
       "              <span class=\"select-menu-item-text css-truncate-target\" title=\"master\">\n",
       "                master\n",
       "              </span>\n",
       "            </a>\n",
       "        </div>\n",
       "\n",
       "          <div class=\"select-menu-no-results\">Nothing to show</div>\n",
       "      </div>\n",
       "\n",
       "      <div class=\"select-menu-list select-menu-tab-bucket js-select-menu-tab-bucket\" data-tab-filter=\"tags\">\n",
       "        <div data-filterable-for=\"context-commitish-filter-field\" data-filterable-type=\"substring\">\n",
       "\n",
       "\n",
       "        </div>\n",
       "\n",
       "        <div class=\"select-menu-no-results\">Nothing to show</div>\n",
       "      </div>\n",
       "\n",
       "    </div>\n",
       "  </div>\n",
       "</div>\n",
       "\n",
       "\n",
       "  <div class=\"breadcrumb\"><span class=\"repo-root js-repo-root\"><span itemscope=\"\" itemtype=\"http://data-vocabulary.org/Breadcrumb\"><a href=\"/dib-lab/2015-petMarSB\" class=\"\" data-branch=\"master\" data-pjax=\"true\" itemscope=\"url\"><span itemprop=\"title\">2015-petMarSB</span></a></span></span><span class=\"separator\">/</span>\n",
       "  <span class=\"tooltipped tooltipped-e disabled text-muted\" aria-label=\"You must be signed in to make or propose changes\">\n",
       "    <span class=\"octicon octicon-plus\"></span>\n",
       "  </span>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "\n",
       "\n",
       "  <div class=\"branch-infobar\">\n",
       "    <span class=\"right\">\n",
       "        <a class=\"muted-link\" href=\"/dib-lab/2015-petMarSB/pull/new/master\">\n",
       "          <span class=\"octicon octicon-git-pull-request\"></span>\n",
       "          Pull request\n",
       "        </a>\n",
       "      <a class=\"muted-link\" href=\"/dib-lab/2015-petMarSB/compare\">\n",
       "        <span class=\"octicon octicon-diff\"></span>\n",
       "        Compare\n",
       "      </a>\n",
       "    </span>\n",
       "\n",
       "      This branch is 64 commits ahead, 1 commit behind camillescott:master.\n",
       "  </div>\n",
       "\n",
       "\n",
       "  <div class=\"commit commit-tease js-details-container\" >\n",
       "    <p class=\"commit-title \">\n",
       "        <a href=\"/dib-lab/2015-petMarSB/commit/9fbc9c1f9bab4e7bef43fe83b75c7d12878814a2\" class=\"message\" data-pjax=\"true\" title=\"Add to_latex function to buscotools\">Add to_latex function to buscotools</a>\n",
       "        \n",
       "    </p>\n",
       "    <div class=\"commit-meta\">\n",
       "      <button aria-label=\"Copy SHA\" class=\"js-zeroclipboard zeroclipboard-link tooltipped tooltipped-s\" data-clipboard-text=\"9fbc9c1f9bab4e7bef43fe83b75c7d12878814a2\" data-copied-hint=\"Copied!\" type=\"button\"><span class=\"octicon octicon-clippy\"></span></button>\n",
       "      <a href=\"/dib-lab/2015-petMarSB/commit/9fbc9c1f9bab4e7bef43fe83b75c7d12878814a2\" class=\"sha-block\" data-pjax>latest commit <span class=\"sha\">9fbc9c1f9b</span></a>\n",
       "\n",
       "      <div class=\"authorship\">\n",
       "        <img alt=\"@camillescott\" class=\"avatar\" height=\"20\" src=\"https://avatars0.githubusercontent.com/u/2896301?v=3&amp;s=40\" width=\"20\" />\n",
       "        <span class=\"author-name\"><a href=\"/camillescott\" rel=\"contributor\">camillescott</a></span>\n",
       "        authored <time class=\"updated\" datetime=\"2015-09-15T00:14:18Z\" is=\"relative-time\">Sep 14, 2015</time>\n",
       "\n",
       "      </div>\n",
       "    </div>\n",
       "  </div>\n",
       "\n",
       "\n",
       "<div class=\"file-wrap\">\n",
       "  <a href=\"/dib-lab/2015-petMarSB/tree/9fbc9c1f9bab4e7bef43fe83b75c7d12878814a2\" class=\"hidden js-permalink-shortcut\" data-hotkey=\"y\">Permalink</a>\n",
       "\n",
       "  <table class=\"files js-navigation-container js-active-navigation-container\" data-pjax>\n",
       "\n",
       "\n",
       "    <tbody>\n",
       "      <tr class=\"warning include-fragment-error\">\n",
       "        <td class=\"icon\"><span class=\"octicon octicon-alert\"></span></td>\n",
       "        <td class=\"content\" colspan=\"3\">Failed to load latest commit information.</td>\n",
       "      </tr>\n",
       "\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-directory\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/tree/master/doc\" class=\"js-directory-link js-navigation-open\" id=\"9a09b4dfda82e3e665e31092d1c3ec8d-12e24bde694f078f5751b6a6e6016ef42f6da3ce\" title=\"doc\">doc</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/d21a63f7c852d2988bac9ca23b591f76871a8a0b\" class=\"message\" data-pjax=\"true\" title=\"update to doc\">update to doc</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-09-10T05:27:30Z\" is=\"time-ago\">Sep 9, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-directory\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/tree/master/germ_dna\" class=\"js-directory-link js-navigation-open\" id=\"4a082270f923fe8d608c169c58fbce93-a2a11d1884e7f8888248c92ac181152a2a45e986\" title=\"germ_dna\">germ_dna</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/06ed18ed7de4dde841f1eb64f61937047e88dee1\" class=\"message\" data-pjax=\"true\" title=\"Add abyss format func to pipeline\">Add abyss format func to pipeline</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-08-22T04:11:16Z\" is=\"time-ago\">Aug 22, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-directory\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/tree/master/notebooks\" class=\"js-directory-link js-navigation-open\" id=\"c2365e874ecb861a659e5215ff17c338-02cbe74def9bdee67c5e9abf0581c01497a33a88\" title=\"notebooks\">notebooks</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/7fd8566e14944bf2729811e7d49f872eae98eca8\" class=\"message\" data-pjax=\"true\" title=\"moore slides\">moore slides</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-09-14T21:02:27Z\" is=\"time-ago\">Sep 14, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-directory\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/tree/master/test\" class=\"js-directory-link js-navigation-open\" id=\"098f6bcd4621d373cade4e832627b4f6-529a3544cffcd0f585c9e05112f280912cff25db\" title=\"test\">test</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/edd65f932146dd30312247feb510e59189bb2608\" class=\"message\" data-pjax=\"true\" title=\"Major refactor of pipeline; removed all peasoup dependencies except BlastTask and got rid of confusing and convoluted pandas apply functions in favor of straight-forward loops.\">Major refactor of pipeline; removed all peasoup dependencies except Bâ¦</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-09-04T00:13:26Z\" is=\"time-ago\">Sep 3, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-text\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/blob/master/.gitignore\" class=\"js-directory-link js-navigation-open\" id=\"a084b794bc0759e7a6b77810e01874f2-91b3de71a24c1bc17206135e11343f5c14aaa692\" title=\".gitignore\">.gitignore</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/8f6b9d4faeae0c5fa77c2ccafbe85d645e93e5ca\" class=\"message\" data-pjax=\"true\" title=\"Add PBS account and user info to templates; add ABySS template and task to pipeline\">Add PBS account and user info to templates; add ABySS template and taâ¦</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-08-22T03:55:11Z\" is=\"time-ago\">Aug 21, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-text\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/blob/master/AUTHORS\" class=\"js-directory-link js-navigation-open\" id=\"3d350169560e75d0cf9fc8e3574a3639-0ffd3ec2eb84408c565ad68234ee66541b38790a\" title=\"AUTHORS\">AUTHORS</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/ecf1f02d73c9b4f36bacaa5ab2b46cc63715c299\" class=\"message\" data-pjax=\"true\" title=\"Add README, AUTHORS\">Add README, AUTHORS</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-04-02T10:12:53Z\" is=\"time-ago\">Apr 2, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-text\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/blob/master/Makefile\" class=\"js-directory-link js-navigation-open\" id=\"b67911656ef5d18c4ae36cb6741b7965-43ab0d2db676c5f9a920e76240850aa699dbe37d\" title=\"Makefile\">Makefile</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/d7c6b38575db66ec418d9aa2bc94e3562cf954f7\" class=\"message\" data-pjax=\"true\" title=\"Removed clean target from Makefile (too easy to hit on accident)\">Removed clean target from Makefile (too easy to hit on accident)</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-09-15T00:08:19Z\" is=\"time-ago\">Sep 14, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-text\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/blob/master/README.md\" class=\"js-directory-link js-navigation-open\" id=\"04c6e90faac2675aa89e2176d2eec7d8-c85455059c56984264a0430f4bb42d2908b33918\" title=\"README.md\">README.md</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/2db158941066534332c0d771d9f96a2904faab5c\" class=\"message\" data-pjax=\"true\" title=\"Some updates to README: authors\">Some updates to README: authors</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-06-04T00:48:46Z\" is=\"time-ago\">Jun 3, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-text\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/blob/master/annotate.py\" class=\"js-directory-link js-navigation-open\" id=\"7d7d020803edf4ebf6306265271a69de-6ce5061a2ce889203be33fe9f89ae1a2d3de0740\" title=\"annotate.py\">annotate.py</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/78c7fa59952fc9bbffbca2ac4b078297530cab02\" class=\"message\" data-pjax=\"true\" title=\"Finish annotation agg task\">Finish annotation agg task</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-08-31T22:01:38Z\" is=\"time-ago\">Aug 31, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-text\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/blob/master/blasttools.py\" class=\"js-directory-link js-navigation-open\" id=\"6746f1127a1cc0a93d53ddb639a275bb-c26fc455da9cdeba24bb484b683597ea81fea363\" title=\"blasttools.py\">blasttools.py</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/78c7fa59952fc9bbffbca2ac4b078297530cab02\" class=\"message\" data-pjax=\"true\" title=\"Finish annotation agg task\">Finish annotation agg task</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-08-31T22:01:38Z\" is=\"time-ago\">Aug 31, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-text\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/blob/master/buscotools.py\" class=\"js-directory-link js-navigation-open\" id=\"11ce44b6793ea844e60287689c155503-f0f3649841ed3b755a388afc6309ab8e74d6008f\" title=\"buscotools.py\">buscotools.py</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/9fbc9c1f9bab4e7bef43fe83b75c7d12878814a2\" class=\"message\" data-pjax=\"true\" title=\"Add to_latex function to buscotools\">Add to_latex function to buscotools</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-09-15T00:14:18Z\" is=\"time-ago\">Sep 14, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-text\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/blob/master/config-test.json\" class=\"js-directory-link js-navigation-open\" id=\"54eb27e47860e1e983c5f70cd592c59a-6a0479e3478a609984c0b4695b4594926798be8a\" title=\"config-test.json\">config-test.json</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/d55b4ebba18a3a9aef9d0588e510945480abeb3d\" class=\"message\" data-pjax=\"true\" title=\"Add first steps of preprocessing pipeline to new script\">Add first steps of preprocessing pipeline to new script</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-06-25T22:56:00Z\" is=\"time-ago\">Jun 25, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-text\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/blob/master/config.json\" class=\"js-directory-link js-navigation-open\" id=\"06b2d3b23dce96e1619d2b53d6c947ec-e6c9dcae698d04258de07d2960a3657f619d5b06\" title=\"config.json\">config.json</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/6aefef941cc79894f623ca2ff571c0094b358269\" class=\"message\" data-pjax=\"true\" title=\"Update BUSCO tasks\">Update BUSCO tasks</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-09-08T04:43:53Z\" is=\"time-ago\">Sep 7, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-text\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/blob/master/environment.conda\" class=\"js-directory-link js-navigation-open\" id=\"4e5eb7cf66dfa8a8b1a88273e116f7b3-33391178743ce6f22c00e3fac2c09f511ca7ecd9\" title=\"environment.conda\">environment.conda</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/cdb0d0e84e97c468d21af3aa9b4cdfb08e88f2e5\" class=\"message\" data-pjax=\"true\" title=\"Update lib location\">Update lib location</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-09-14T19:44:17Z\" is=\"time-ago\">Sep 14, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-text\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/blob/master/gtftools.py\" class=\"js-directory-link js-navigation-open\" id=\"a288226bc72fd3d850e2af620bf9a4fa-6490126110cc1e01c42f1c083073ab20aba0739a\" title=\"gtftools.py\">gtftools.py</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/78c7fa59952fc9bbffbca2ac4b078297530cab02\" class=\"message\" data-pjax=\"true\" title=\"Finish annotation agg task\">Finish annotation agg task</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-08-31T22:01:38Z\" is=\"time-ago\">Aug 31, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-text\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/blob/master/hmmertools.py\" class=\"js-directory-link js-navigation-open\" id=\"34d6ac2fd79876571b1a3770ccdada70-26ff442bb3854bacfed3bfecbdadd098275f5db7\" title=\"hmmertools.py\">hmmertools.py</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/78c7fa59952fc9bbffbca2ac4b078297530cab02\" class=\"message\" data-pjax=\"true\" title=\"Finish annotation agg task\">Finish annotation agg task</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-08-31T22:01:38Z\" is=\"time-ago\">Aug 31, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-text\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/blob/master/pipeline\" class=\"js-directory-link js-navigation-open\" id=\"99f6f4be0908f24bb4a22a4ffb277da4-219ab1e43c225a8ce5b1c69b7d9463f58fb49bcf\" title=\"pipeline\">pipeline</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/6aefef941cc79894f623ca2ff571c0094b358269\" class=\"message\" data-pjax=\"true\" title=\"Update BUSCO tasks\">Update BUSCO tasks</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-09-08T04:43:53Z\" is=\"time-ago\">Sep 8, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-text\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/blob/master/remap_blast.pyx\" class=\"js-directory-link js-navigation-open\" id=\"5fb87f44ad197d4a606970f3f09f7aef-b96ecaa726234075b4146d50be067c1878397912\" title=\"remap_blast.pyx\">remap_blast.pyx</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/78c7fa59952fc9bbffbca2ac4b078297530cab02\" class=\"message\" data-pjax=\"true\" title=\"Finish annotation agg task\">Finish annotation agg task</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-08-31T22:01:38Z\" is=\"time-ago\">Aug 31, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-text\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/blob/master/resources-test.json\" class=\"js-directory-link js-navigation-open\" id=\"fa8720e9b46660cd48f02be5166cab8f-2acd992fdf9bf452bad2b02d3ac39ba22d7ced0b\" title=\"resources-test.json\">resources-test.json</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/34d371358a22283acb2ef8d3555a5ec83af035d4\" class=\"message\" data-pjax=\"true\" title=\"Slight update to metadata formatting\">Slight update to metadata formatting</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-05-13T21:39:09Z\" is=\"time-ago\">May 13, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-text\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/blob/master/resources.json\" class=\"js-directory-link js-navigation-open\" id=\"a031fd230f8287e80b8c9bbfb195986f-f3d001e8d291d9db3b17c641d04a37b0ebf3998a\" title=\"resources.json\">resources.json</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/6aefef941cc79894f623ca2ff571c0094b358269\" class=\"message\" data-pjax=\"true\" title=\"Update BUSCO tasks\">Update BUSCO tasks</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-09-08T04:43:53Z\" is=\"time-ago\">Sep 8, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "        <tr class=\"js-navigation-item\">\n",
       "          <td class=\"icon\">\n",
       "            <span class=\"octicon octicon-file-text\"></span>\n",
       "            <img alt=\"\" class=\"spinner\" height=\"16\" src=\"https://assets-cdn.github.com/images/spinners/octocat-spinner-32.gif\" width=\"16\" />\n",
       "          </td>\n",
       "          <td class=\"content\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><a href=\"/dib-lab/2015-petMarSB/blob/master/tasks.py\" class=\"js-directory-link js-navigation-open\" id=\"dc31105b816206de63c47bf42ec2a4d7-f288506399f51777a78236da2f1c2876a127dae9\" title=\"tasks.py\">tasks.py</a></span>\n",
       "          </td>\n",
       "          <td class=\"message\">\n",
       "            <span class=\"css-truncate css-truncate-target\">\n",
       "              <a href=\"/dib-lab/2015-petMarSB/commit/9fbc9c1f9bab4e7bef43fe83b75c7d12878814a2\" class=\"message\" data-pjax=\"true\" title=\"Add to_latex function to buscotools\">Add to_latex function to buscotools</a>\n",
       "            </span>\n",
       "          </td>\n",
       "          <td class=\"age\">\n",
       "            <span class=\"css-truncate css-truncate-target\"><time datetime=\"2015-09-15T00:14:18Z\" is=\"time-ago\">Sep 15, 2015</time></span>\n",
       "          </td>\n",
       "        </tr>\n",
       "    </tbody>\n",
       "  </table>\n",
       "\n",
       "</div>\n",
       "\n",
       "\n",
       "  <div id=\"readme\" class=\"boxed-group flush clearfix announce instapaper_body md\">\n",
       "    <h3>\n",
       "      <span class=\"octicon octicon-book\"></span>\n",
       "      README.md\n",
       "    </h3>\n",
       "\n",
       "    <article class=\"markdown-body entry-content\" itemprop=\"mainContentOfPage\"><h1><a id=\"user-content-2015-petmarsb\" class=\"anchor\" href=\"#2015-petmarsb\" aria-hidden=\"true\"><span class=\"octicon octicon-link\"></span></a>2015-petMarSB</h1>\n",
       "\n",
       "<p>This is the new home of the lamprey transcriptome sequencing and assembly project. Contains pipelines for preprocessing data, running the assembly, and postprocressing that assembly, IPython notebooks for analyzing the results. This is a cleaned up and more production-ready version of the work; previous (and messier) work is available at <a href=\"https://github.com/camillescott/2013-lamprey\">https://github.com/camillescott/2013-lamprey</a></p>\n",
       "\n",
       "<h2><a id=\"user-content-getting-started\" class=\"anchor\" href=\"#getting-started\" aria-hidden=\"true\"><span class=\"octicon octicon-link\"></span></a>Getting Started</h2>\n",
       "\n",
       "<p></p>\n",
       "\n",
       "<h2><a id=\"user-content-authors\" class=\"anchor\" href=\"#authors\" aria-hidden=\"true\"><span class=\"octicon octicon-link\"></span></a>Authors</h2>\n",
       "\n",
       "<p>Code for this repo is written/maintained (for some values of maintained) by Camille Scott. See the AUTHORS file for details and additional collaborators.</p>\n",
       "\n",
       "<h2><a id=\"user-content-license\" class=\"anchor\" href=\"#license\" aria-hidden=\"true\"><span class=\"octicon octicon-link\"></span></a>License</h2>\n",
       "\n",
       "<p>The work in this repository is CC-BY (Creative Commons Attribution 3.0) licensed. This means it can be copied, redistributed, remixed, transformed, and otherwised used or tinkered with however you see fit, provided that you attribute the author (Camille Scott) appropriately, provide a link to the license, and note if changes were made. License information is available at <a href=\"https://creativecommons.org/licenses/by/3.0/us/\">https://creativecommons.org/licenses/by/3.0/us/</a>.</p>\n",
       "</article>\n",
       "  </div>\n",
       "\n",
       "\n",
       "        </div>\n",
       "      </div>\n",
       "      <div class=\"modal-backdrop\"></div>\n",
       "    </div>\n",
       "  </div>\n",
       "\n",
       "\n",
       "    </div>\n",
       "\n",
       "      <div class=\"container\">\n",
       "  <div class=\"site-footer\" role=\"contentinfo\">\n",
       "    <ul class=\"site-footer-links right\">\n",
       "        <li><a href=\"https://status.github.com/\" data-ga-click=\"Footer, go to status, text:status\">Status</a></li>\n",
       "      <li><a href=\"https://developer.github.com\" data-ga-click=\"Footer, go to api, text:api\">API</a></li>\n",
       "      <li><a href=\"https://training.github.com\" data-ga-click=\"Footer, go to training, text:training\">Training</a></li>\n",
       "      <li><a href=\"https://shop.github.com\" data-ga-click=\"Footer, go to shop, text:shop\">Shop</a></li>\n",
       "        <li><a href=\"https://github.com/blog\" data-ga-click=\"Footer, go to blog, text:blog\">Blog</a></li>\n",
       "        <li><a href=\"https://github.com/about\" data-ga-click=\"Footer, go to about, text:about\">About</a></li>\n",
       "        <li><a href=\"https://github.com/pricing\" data-ga-click=\"Footer, go to pricing, text:pricing\">Pricing</a></li>\n",
       "\n",
       "    </ul>\n",
       "\n",
       "    <a href=\"https://github.com\" aria-label=\"Homepage\">\n",
       "      <span class=\"mega-octicon octicon-mark-github\" title=\"GitHub\"></span>\n",
       "</a>\n",
       "    <ul class=\"site-footer-links\">\n",
       "      <li>&copy; 2015 <span title=\"0.08215s from github-fe127-cp1-prd.iad.github.net\">GitHub</span>, Inc.</li>\n",
       "        <li><a href=\"https://github.com/site/terms\" data-ga-click=\"Footer, go to terms, text:terms\">Terms</a></li>\n",
       "        <li><a href=\"https://github.com/site/privacy\" data-ga-click=\"Footer, go to privacy, text:privacy\">Privacy</a></li>\n",
       "        <li><a href=\"https://github.com/security\" data-ga-click=\"Footer, go to security, text:security\">Security</a></li>\n",
       "        <li><a href=\"https://github.com/contact\" data-ga-click=\"Footer, go to contact, text:contact\">Contact</a></li>\n",
       "        <li><a href=\"https://help.github.com\" data-ga-click=\"Footer, go to help, text:help\">Help</a></li>\n",
       "    </ul>\n",
       "  </div>\n",
       "</div>\n",
       "\n",
       "\n",
       "\n",
       "    \n",
       "    \n",
       "\n",
       "    <div id=\"ajax-error-message\" class=\"flash flash-error\">\n",
       "      <span class=\"octicon octicon-alert\"></span>\n",
       "      <button type=\"button\" class=\"flash-close js-flash-close js-ajax-error-dismiss\" aria-label=\"Dismiss error\">\n",
       "        <span class=\"octicon octicon-x\"></span>\n",
       "      </button>\n",
       "      Something went wrong with that request. Please try again.\n",
       "    </div>\n",
       "\n",
       "\n",
       "      <script crossorigin=\"anonymous\" src=\"https://assets-cdn.github.com/assets/frameworks-2c67e846b237d12dc49d9ba38bef688797518db6c4fc7ec256caf871623c69e4.js\"></script>\n",
       "      <script async=\"async\" crossorigin=\"anonymous\" src=\"https://assets-cdn.github.com/assets/github-cdf2812e9b014c7112138481f998b4d18770faa142b62ce3c3223fbf481b6a94.js\"></script>\n",
       "      \n",
       "      \n",
       "    <div class=\"js-stale-session-flash stale-session-flash flash flash-warn flash-banner hidden\">\n",
       "      <span class=\"octicon octicon-alert\"></span>\n",
       "      <span class=\"signed-in-tab-flash\">You signed in with another tab or window. <a href=\"\">Reload</a> to refresh your session.</span>\n",
       "      <span class=\"signed-out-tab-flash\">You signed out in another tab or window. <a href=\"\">Reload</a> to refresh your session.</span>\n",
       "    </div>\n",
       "  </body>\n",
       "</html>\n",
       "\n"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "HTML('https://github.com/dib-lab/2015-petMarSB')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "slideshow": {
     "slide_type": "subslide"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "<!DOCTYPE html>\n",
       "<html lang=\"en\">\n",
       "\n",
       "<head>\n",
       "  <meta charset=\"utf-8\">\n",
       "  <title></title>\n",
       "  <meta name=\"viewport\" content=\"width=device-width, initial-scale=1.0\">\n",
       "  <meta name=\"description\" content=\"\">\n",
       "  <meta name=\"author\" content=\"\">\n",
       "  \n",
       "  <meta name=\"robots\" content=\"noindex,nofollow\">\n",
       "  \n",
       "\n",
       "  <!--NEW RELIC Start Perf Measurement-->\n",
       "  \n",
       "  <!--NREND-->\n",
       "\n",
       "  <!-- Le styles -->\n",
       "  <link href=\"/static/build/styles.css\" rel=\"stylesheet\">\n",
       "\n",
       "  <!-- Le HTML5 shim, for IE6-8 support of HTML5 elements -->\n",
       "  <!--[if lt IE 9]>\n",
       "    <script src=\"http://html5shim.googlecode.com/svn/trunk/html5.js\"></script>\n",
       "  <![endif]-->\n",
       "\n",
       "  <!-- Le fav and touch icons -->\n",
       "  <link rel=\"shortcut icon\" href=\"/static/ico/ipynb_icon_16x16.png\">\n",
       "  <link rel=\"apple-touch-icon-precomposed\" sizes=\"144x144\" \n",
       "        href=\"/static/ico/apple-touch-icon-144-precomposed.png\">\n",
       "  <link rel=\"apple-touch-icon-precomposed\" sizes=\"114x114\" \n",
       "        href=\"/static/ico/apple-touch-icon-114-precomposed.png\">\n",
       "  <link rel=\"apple-touch-icon-precomposed\" sizes=\"72x72\" \n",
       "        href=\"/static/ico/apple-touch-icon-72-precomposed.png\">\n",
       "  <link rel=\"apple-touch-icon-precomposed\" \n",
       "        href=\"/static/ico/apple-touch-icon-57-precomposed.png\">\n",
       "  \n",
       "  \n",
       "\n",
       "  \n",
       "\n",
       "  \n",
       "  <meta name=\"twitter:card\" content=\"summary\">\n",
       "  <meta name=\"twitter:title\" content=\"Notebook on nbviewer\">\n",
       "  <meta name=\"twitter:description\" content=\"Check out this Jupyter notebook!\">\n",
       "\n",
       "  \n",
       "  <meta name=\"twitter:domain\" content=\"nbviewer.ipython.org\">\n",
       "  <meta name=\"twitter:image:src\" content=\"http://ipython.org/ipython-doc/dev/_images/ipynb_icon_128x128.png\">\n",
       "\n",
       "  \n",
       "    <link href=\"/static/build/notebook.css\" rel=\"stylesheet\">\n",
       "  \n",
       "\n",
       "  \n",
       "\n",
       "  \n",
       "    <script src=\"https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML\" type=\"text/javascript\">\n",
       "    </script>\n",
       "    <script type=\"text/javascript\">\n",
       "      init_mathjax = function() {\n",
       "        if (window.MathJax) {\n",
       "          // MathJax loaded\n",
       "          MathJax.Hub.Config({\n",
       "            TeX: {\n",
       "              equationNumbers: {\n",
       "                autoNumber: \"AMS\",\n",
       "                useLabelIds: true\n",
       "              }\n",
       "            },\n",
       "            tex2jax: {\n",
       "              inlineMath: [ ['$','$'], [\"\\\\(\",\"\\\\)\"] ],\n",
       "              displayMath: [ ['$$','$$'], [\"\\\\[\",\"\\\\]\"] ],\n",
       "              processEscapes: true,\n",
       "              processEnvironments: true\n",
       "            },\n",
       "            displayAlign: 'center',\n",
       "            \"HTML-CSS\": {\n",
       "              styles: {'.MathJax_Display': {\"margin\": 0}},\n",
       "              linebreaks: { automatic: true }\n",
       "            }\n",
       "          });\n",
       "          MathJax.Hub.Queue([\"Typeset\", MathJax.Hub]);\n",
       "        }\n",
       "      }\n",
       "      init_mathjax();\n",
       "    </script>\n",
       "  \n",
       "\n",
       "</head>\n",
       "\n",
       "<body data-spy=\"scroll\" data-target=\".subnav\" data-offset=\"50\">\n",
       "\n",
       "  <!-- These are loaded at the top of the body so they are available to \n",
       "       notebook cells when they are loaded below. -->\n",
       "  <script src=\"/static/components/jquery/dist/jquery.min.js\"></script>\n",
       "  <script src=\"/static/components/requirejs/require.js\"></script>\n",
       "  <script src=\"/static/components/moment/min/moment.min.js\"></script>\n",
       "\n",
       "<!-- Navbar\n",
       "================================================== -->\n",
       "  <nav id=\"menubar\" class=\"navbar navbar-default navbar-fixed-top\">\n",
       "    <div class=\"container\">\n",
       "      <div class=\"navbar-header\">\n",
       "        <button type=\"button\" class=\"navbar-toggle collapsed\" data-toggle=\"collapse\" data-target=\".navbar-collapse\">\n",
       "          <span class=\"sr-only\">Toggle navigation</span>\n",
       "          <i class=\"fa fa-bars\"></i>\n",
       "        </button>\n",
       "        <a class=\"navbar-brand\" href=\"/\">nbviewer</a>\n",
       "      </div>\n",
       "\n",
       "      <div class=\"collapse navbar-collapse\">\n",
       "        <ul class=\"nav navbar-nav\">\n",
       "          <li>\n",
       "    <a href=\"/faq\" title=\"FAQ\" >\n",
       "      \n",
       "        <span>FAQ</span>\n",
       "      \n",
       "    </a>\n",
       "  </li>\n",
       "          <li class=\"\">\n",
       "            <a href=\"http://ipython.org\">IPython</a>\n",
       "          </li>\n",
       "          <li class=\"\">\n",
       "            <a href=\"http://jupyter.org\">Jupyter</a>\n",
       "          </li>\n",
       "        </ul>\n",
       "        \n",
       "        <ul class=\"nav navbar-nav navbar-right\">\n",
       "          \n",
       "  \n",
       "    \n",
       "  \n",
       "\n",
       "  \n",
       "    <li>\n",
       "    <a href=\"#\" title=\"Python 2 Kernel\" >\n",
       "      <span class=\"fa fa-server fa-2x menu-icon\"></span>\n",
       "      <span class=\"menu-text\">Python 2 Kernel</span>\n",
       "    </a>\n",
       "  </li>\n",
       "  \n",
       "  \n",
       "  \n",
       "    <li>\n",
       "    <a href=\"https://github.com/dib-lab/2015-petMarSB/blob/master/notebooks/petmar-genome-completeness.ipynb\" title=\"View on GitHub\" >\n",
       "      <span class=\"fa fa-github fa-2x menu-icon\"></span>\n",
       "      <span class=\"menu-text\">View on GitHub</span>\n",
       "    </a>\n",
       "  </li>\n",
       "  \n",
       "\n",
       "  <li>\n",
       "    <a href=\"https://raw.githubusercontent.com/dib-lab/2015-petMarSB/master/notebooks/petmar-genome-completeness.ipynb\" title=\"Download Notebook\" download>\n",
       "      <span class=\"fa fa-download fa-2x menu-icon\"></span>\n",
       "      <span class=\"menu-text\">Download Notebook</span>\n",
       "    </a>\n",
       "  </li>\n",
       "\n",
       "        </ul>\n",
       "      </div><!-- /.navbar-collapse -->\n",
       "      \n",
       "      \n",
       "    </div>\n",
       "  </nav>\n",
       "\n",
       "  <div class=\"container\">\n",
       "    \n",
       "  \n",
       "  <ol class=\"breadcrumb\">\n",
       "    \n",
       "      <li>\n",
       "        <a href=\"/github/dib-lab/2015-petMarSB/tree/master\">2015-petMarSB</a>\n",
       "      </li>\n",
       "    \n",
       "      <li>\n",
       "        <a href=\"/github/dib-lab/2015-petMarSB/tree/master/notebooks\">notebooks</a>\n",
       "      </li>\n",
       "    \n",
       "  </ol>\n",
       "  \n",
       "  <div id=\"notebook\">\n",
       "    <div id=\"notebook-container\">\n",
       "      \n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<h1 id=\"Lamprey-Transcriptome-Analysis:-Genome-Completeness-Assessment\">Lamprey Transcriptome Analysis: Genome Completeness Assessment<a class=\"anchor-link\" href=\"#Lamprey-Transcriptome-Analysis:-Genome-Completeness-Assessment\">&#182;</a></h1>\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "\n",
       "<pre><code>Camille Scott [camille dot scott dot w @gmail.com] [@camille_codon]\n",
       "\n",
       "camillescott.github.io\n",
       "\n",
       "Lab for Data Intensive Biology (DIB)\n",
       "University of California Davis</code></pre>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<h2 id=\"About\">About<a class=\"anchor-link\" href=\"#About\">&#182;</a></h2>\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<p>Uses the many-sample de novo transcriptome assembly to attempt to assess the completeness of the genome build (and orthogonally, the completeness of the transcriptome itself).</p>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "\n",
       "<pre><code>assembly version: lamp10\n",
       "\n",
       "assembly program: Trinity\n",
       "\n",
       "genome build: 7.0.75 (ensembl release 75</code></pre>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<h2 id=\"Libraries\">Libraries<a class=\"anchor-link\" href=\"#Libraries\">&#182;</a></h2>\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[3]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"o\">%</span><span class=\"k\">load_ext</span> autoreload\n",
       "<span class=\"o\">%</span><span class=\"k\">autoreload</span> 2\n",
       "<span class=\"o\">%</span><span class=\"k\">autosave</span> 60\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt\"></div>\n",
       "\n",
       "\n",
       "<div class=\"output_subarea output_javascript \">\n",
       "<script type=\"text/javascript\">\n",
       "IPython.notebook.set_autosave_interval(60000)\n",
       "</script>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt\"></div>\n",
       "<div class=\"output_subarea output_stream output_stdout output_text\">\n",
       "<pre>Autosaving every 60 seconds\n",
       "</pre>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[4]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"kn\">from</span> <span class=\"nn\">libs</span> <span class=\"kn\">import</span> <span class=\"o\">*</span>\n",
       "<span class=\"o\">%</span><span class=\"k\">run</span> -i common.ipy\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt\"></div>\n",
       "<div class=\"output_subarea output_stream output_stderr output_text\">\n",
       "<pre>** Using data resources found in ../resources.json\n",
       "** Using config found in ../config.json\n",
       "</pre>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[5]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"kn\">import</span> <span class=\"nn\">pyprind</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[6]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"kn\">from</span> <span class=\"nn\">blasttools</span> <span class=\"kn\">import</span> <span class=\"n\">blast_to_df</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[7]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"kn\">from</span> <span class=\"nn\">IPython.display</span> <span class=\"kn\">import</span> <span class=\"n\">display</span><span class=\"p\">,</span> <span class=\"n\">Image</span>\n",
       "<span class=\"kn\">import</span> <span class=\"nn\">glob</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[8]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"o\">%</span><span class=\"k\">pylab</span> inline\n",
       "<span class=\"kn\">from</span> <span class=\"nn\">matplotlib</span> <span class=\"kn\">import</span> <span class=\"n\">rc_context</span>\n",
       "<span class=\"n\">tall_size</span> <span class=\"o\">=</span> <span class=\"p\">(</span><span class=\"mi\">8</span><span class=\"p\">,</span><span class=\"mi\">16</span><span class=\"p\">)</span>\n",
       "<span class=\"n\">norm_size</span> <span class=\"o\">=</span> <span class=\"p\">(</span><span class=\"mi\">12</span><span class=\"p\">,</span><span class=\"mi\">8</span><span class=\"p\">)</span>\n",
       "<span class=\"n\">mpl_params</span> <span class=\"o\">=</span> <span class=\"p\">{</span><span class=\"s\">&#39;figure.autolayout&#39;</span><span class=\"p\">:</span> <span class=\"bp\">True</span><span class=\"p\">,</span>\n",
       "               <span class=\"s\">&#39;axes.titlesize&#39;</span><span class=\"p\">:</span> <span class=\"mi\">24</span><span class=\"p\">,</span>\n",
       "               <span class=\"s\">&#39;axes.labelsize&#39;</span><span class=\"p\">:</span> <span class=\"mi\">16</span><span class=\"p\">,</span>\n",
       "               <span class=\"s\">&#39;ytick.labelsize&#39;</span><span class=\"p\">:</span> <span class=\"mi\">14</span><span class=\"p\">,</span>\n",
       "               <span class=\"s\">&#39;xtick.labelsize&#39;</span><span class=\"p\">:</span> <span class=\"mi\">14</span>\n",
       "               <span class=\"p\">}</span>\n",
       "<span class=\"n\">sns</span><span class=\"o\">.</span><span class=\"n\">set</span><span class=\"p\">(</span><span class=\"n\">style</span><span class=\"o\">=</span><span class=\"s\">&quot;white&quot;</span><span class=\"p\">,</span> <span class=\"n\">palette</span><span class=\"o\">=</span><span class=\"s\">&quot;Paired&quot;</span><span class=\"p\">,</span> <span class=\"n\">rc</span><span class=\"o\">=</span><span class=\"n\">mpl_params</span><span class=\"p\">)</span>\n",
       "<span class=\"c\">#sns.set_palette(&quot;Paired&quot;, desat=.6)</span>\n",
       "<span class=\"n\">b</span><span class=\"p\">,</span> <span class=\"n\">g</span><span class=\"p\">,</span> <span class=\"n\">r</span><span class=\"p\">,</span> <span class=\"n\">p</span> <span class=\"o\">=</span> <span class=\"n\">sns</span><span class=\"o\">.</span><span class=\"n\">color_palette</span><span class=\"p\">(</span><span class=\"s\">&quot;muted&quot;</span><span class=\"p\">,</span> <span class=\"mi\">4</span><span class=\"p\">)</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt\"></div>\n",
       "<div class=\"output_subarea output_stream output_stdout output_text\">\n",
       "<pre>Populating the interactive namespace from numpy and matplotlib\n",
       "</pre>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[9]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"kn\">from</span> <span class=\"nn\">ete3</span> <span class=\"kn\">import</span> <span class=\"n\">NCBITaxa</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[10]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"o\">%</span><span class=\"k\">config</span> InlineBackend.close_figures = False\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<h2 id=\"Data\">Data<a class=\"anchor-link\" href=\"#Data\">&#182;</a></h2>\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<p>We'll save our results to a dictionary and dump it to JSON for use in the paper.</p>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[11]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">results</span> <span class=\"o\">=</span> <span class=\"p\">{}</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[12]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">store</span> <span class=\"o\">=</span> <span class=\"n\">pd</span><span class=\"o\">.</span><span class=\"n\">HDFStore</span><span class=\"p\">(</span><span class=\"n\">wdir</span><span class=\"p\">(</span><span class=\"s\">&#39;{}.store.h5&#39;</span><span class=\"o\">.</span><span class=\"n\">format</span><span class=\"p\">(</span><span class=\"n\">prefix</span><span class=\"p\">)),</span> <span class=\"n\">complib</span><span class=\"o\">=</span><span class=\"s\">&#39;zlib&#39;</span><span class=\"p\">,</span> <span class=\"n\">complevel</span><span class=\"o\">=</span><span class=\"mi\">5</span><span class=\"p\">)</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[13]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"kn\">import</span> <span class=\"nn\">atexit</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[14]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"k\">def</span> <span class=\"nf\">dump_results</span><span class=\"p\">(</span><span class=\"n\">fn</span><span class=\"o\">=</span><span class=\"s\">&#39;../doc/petmar-genome-comp.results.json&#39;</span><span class=\"p\">):</span>\n",
       "    <span class=\"k\">with</span> <span class=\"nb\">open</span><span class=\"p\">(</span><span class=\"n\">fn</span><span class=\"p\">,</span> <span class=\"s\">&#39;wb&#39;</span><span class=\"p\">)</span> <span class=\"k\">as</span> <span class=\"n\">fp</span><span class=\"p\">:</span>\n",
       "        <span class=\"n\">json</span><span class=\"o\">.</span><span class=\"n\">dump</span><span class=\"p\">(</span><span class=\"n\">results</span><span class=\"p\">,</span> <span class=\"n\">fp</span><span class=\"p\">)</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[15]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"k\">def</span> <span class=\"nf\">exit_func</span><span class=\"p\">():</span>\n",
       "    <span class=\"n\">dump_results</span><span class=\"p\">()</span>\n",
       "    <span class=\"n\">store</span><span class=\"o\">.</span><span class=\"n\">close</span><span class=\"p\">()</span>\n",
       "<span class=\"n\">atexit</span><span class=\"o\">.</span><span class=\"n\">register</span><span class=\"p\">(</span><span class=\"n\">exit_func</span><span class=\"p\">)</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt output_prompt\">Out[15]:</div>\n",
       "\n",
       "\n",
       "<div class=\"output_text output_subarea output_execute_result\">\n",
       "<pre>&lt;function __main__.exit_func&gt;</pre>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<h2 id=\"Orthologs\">Orthologs<a class=\"anchor-link\" href=\"#Orthologs\">&#182;</a></h2>\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<p>How many transcripts have orthologs in other organisms, but no lamprey support? And vice versa?</p>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[16]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">ortho_panel</span> <span class=\"o\">=</span> <span class=\"n\">store</span><span class=\"p\">[</span><span class=\"s\">&#39;lamp10_ortho&#39;</span><span class=\"p\">]</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[17]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">ortho_df</span> <span class=\"o\">=</span> <span class=\"n\">store</span><span class=\"p\">[</span><span class=\"s\">&#39;lamp10_ortho_filter_df&#39;</span><span class=\"p\">]</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[18]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">blast_df</span> <span class=\"o\">=</span> <span class=\"n\">store</span><span class=\"p\">[</span><span class=\"s\">&#39;lamp10_blast_filter_df&#39;</span><span class=\"p\">]</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[19]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"c\"># Transcripts with recipricol best thits to our main databases</span>\n",
       "<span class=\"n\">has_ortho</span> <span class=\"o\">=</span> <span class=\"p\">((</span><span class=\"n\">ortho_df</span><span class=\"p\">[</span><span class=\"s\">&#39;lamp10.fasta.x.braFlo.pep.all.fa.db.tsv&#39;</span><span class=\"p\">]</span> <span class=\"o\">==</span> <span class=\"bp\">True</span><span class=\"p\">)</span> <span class=\"o\">|</span> \n",
       "             <span class=\"p\">(</span><span class=\"n\">ortho_df</span><span class=\"p\">[</span><span class=\"s\">&#39;lamp10.fasta.x.danRer.pep.fa.db.tsv&#39;</span><span class=\"p\">]</span> <span class=\"o\">==</span> <span class=\"bp\">True</span><span class=\"p\">)</span> <span class=\"o\">|</span>\n",
       "             <span class=\"p\">(</span><span class=\"n\">ortho_df</span><span class=\"p\">[</span><span class=\"s\">&#39;lamp10.fasta.x.homSap.pep.fa.db.tsv&#39;</span><span class=\"p\">]</span> <span class=\"o\">==</span> <span class=\"bp\">True</span><span class=\"p\">)</span> <span class=\"o\">|</span>\n",
       "             <span class=\"p\">(</span><span class=\"n\">ortho_df</span><span class=\"p\">[</span><span class=\"s\">&#39;lamp10.fasta.x.musMus.pep.fa.db.tsv&#39;</span><span class=\"p\">]</span> <span class=\"o\">==</span> <span class=\"bp\">True</span><span class=\"p\">))</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[20]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"c\"># We want to filter out transcripts with *any* homologies to the genome,</span>\n",
       "<span class=\"c\"># the lamprey proteins, and the lamprey cDNAs (lamp0)</span>\n",
       "<span class=\"n\">lamp_filter</span> <span class=\"o\">=</span> <span class=\"p\">((</span><span class=\"n\">blast_df</span><span class=\"p\">[</span><span class=\"s\">&#39;lamp10.fasta.x.petMar2.fa.db.tsv&#39;</span><span class=\"p\">]</span> <span class=\"o\">==</span> <span class=\"bp\">False</span><span class=\"p\">)</span> <span class=\"o\">&amp;</span> \n",
       "               <span class=\"p\">(</span><span class=\"n\">blast_df</span><span class=\"p\">[</span><span class=\"s\">&#39;lamp10.fasta.x.petMar2.pep.fa.db.tsv&#39;</span><span class=\"p\">]</span> <span class=\"o\">==</span> <span class=\"bp\">False</span><span class=\"p\">)</span> <span class=\"o\">&amp;</span>\n",
       "               <span class=\"p\">(</span><span class=\"n\">blast_df</span><span class=\"p\">[</span><span class=\"s\">&#39;lamp10.fasta.x.petMar2.cdna.fa.db.tsv&#39;</span><span class=\"p\">]</span> <span class=\"o\">==</span> <span class=\"bp\">False</span><span class=\"p\">))</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[21]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">ortho_df</span> <span class=\"o\">=</span> <span class=\"n\">ortho_panel</span><span class=\"o\">.</span><span class=\"n\">minor_xs</span><span class=\"p\">(</span><span class=\"s\">&#39;sseqid&#39;</span><span class=\"p\">)</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[22]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">n_novel</span> <span class=\"o\">=</span> <span class=\"p\">(</span><span class=\"n\">has_ortho</span> <span class=\"o\">&amp;</span> <span class=\"n\">lamp_filter</span><span class=\"p\">)</span><span class=\"o\">.</span><span class=\"n\">sum</span><span class=\"p\">()</span>\n",
       "<span class=\"n\">results</span><span class=\"p\">[</span><span class=\"s\">&#39;n_novel_ortho&#39;</span><span class=\"p\">]</span> <span class=\"o\">=</span> <span class=\"n\">n_novel</span>\n",
       "<span class=\"k\">print</span> <span class=\"n\">n_novel</span><span class=\"p\">,</span> <span class=\"s\">&#39;have orthologies but no lamprey support&#39;</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt\"></div>\n",
       "<div class=\"output_subarea output_stream output_stdout output_text\">\n",
       "<pre>1768 have orthologies but no lamprey support\n",
       "</pre>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[23]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">n_genome_supported</span> <span class=\"o\">=</span> <span class=\"p\">(</span><span class=\"n\">has_ortho</span> <span class=\"o\">&amp;</span> <span class=\"p\">(</span><span class=\"n\">blast_df</span><span class=\"p\">[</span><span class=\"s\">&#39;lamp10.fasta.x.petMar2.fa.db.tsv&#39;</span><span class=\"p\">]</span> <span class=\"o\">==</span> <span class=\"bp\">True</span><span class=\"p\">))</span><span class=\"o\">.</span><span class=\"n\">sum</span><span class=\"p\">()</span>\n",
       "<span class=\"n\">results</span><span class=\"p\">[</span><span class=\"s\">&#39;n_genome_supported&#39;</span><span class=\"p\">]</span> <span class=\"o\">=</span> <span class=\"n\">n_genome_supported</span>\n",
       "<span class=\"k\">print</span> <span class=\"n\">n_genome_supported</span><span class=\"p\">,</span> <span class=\"s\">&#39;have orthologies and genome support&#39;</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt\"></div>\n",
       "<div class=\"output_subarea output_stream output_stdout output_text\">\n",
       "<pre>11990 have orthologies and genome support\n",
       "</pre>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[24]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">n_supported</span> <span class=\"o\">=</span> <span class=\"p\">(</span><span class=\"n\">has_ortho</span> <span class=\"o\">&amp;</span> <span class=\"p\">(</span><span class=\"n\">lamp_filter</span> <span class=\"o\">==</span> <span class=\"bp\">False</span><span class=\"p\">))</span><span class=\"o\">.</span><span class=\"n\">sum</span><span class=\"p\">()</span>\n",
       "<span class=\"n\">results</span><span class=\"p\">[</span><span class=\"s\">&#39;n_supported_ortho&#39;</span><span class=\"p\">]</span> <span class=\"o\">=</span> <span class=\"n\">n_supported</span>\n",
       "<span class=\"k\">print</span> <span class=\"n\">n_supported</span><span class=\"p\">,</span> <span class=\"s\">&#39;have orthologies and lamprey support&#39;</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt\"></div>\n",
       "<div class=\"output_subarea output_stream output_stdout output_text\">\n",
       "<pre>13405 have orthologies and lamprey support\n",
       "</pre>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[25]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"k\">print</span> <span class=\"s\">&#39;This means {:.2f}</span><span class=\"si\">% o</span><span class=\"s\">f orthologs are potentially novel&#39;</span><span class=\"o\">.</span><span class=\"n\">format</span><span class=\"p\">((</span><span class=\"nb\">float</span><span class=\"p\">(</span><span class=\"n\">n_novel</span><span class=\"p\">)</span> <span class=\"o\">/</span> <span class=\"p\">(</span><span class=\"n\">n_novel</span> <span class=\"o\">+</span> <span class=\"n\">n_supported</span><span class=\"p\">))</span> <span class=\"o\">*</span> <span class=\"mf\">100.0</span><span class=\"p\">)</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt\"></div>\n",
       "<div class=\"output_subarea output_stream output_stdout output_text\">\n",
       "<pre>This means 11.65% of orthologs are potentially novel\n",
       "</pre>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<h3 id=\"Corresponding-Genes\">Corresponding Genes<a class=\"anchor-link\" href=\"#Corresponding-Genes\">&#182;</a></h3>\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[52]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">GNATHOSTOMATA</span> <span class=\"o\">=</span> <span class=\"mi\">7776</span>\n",
       "<span class=\"n\">CYCLOSTOMATA</span> <span class=\"o\">=</span> <span class=\"mi\">1476529</span>\n",
       "<span class=\"n\">CEPHALOCHORDATA</span> <span class=\"o\">=</span> <span class=\"mi\">7735</span> \n",
       "<span class=\"n\">PETMAR</span> <span class=\"o\">=</span> <span class=\"mi\">7757</span>\n",
       "<span class=\"n\">BRAFLO</span> <span class=\"o\">=</span> <span class=\"mi\">7739</span>\n",
       "<span class=\"n\">MUSMUS</span> <span class=\"o\">=</span> <span class=\"mi\">10090</span>\n",
       "<span class=\"n\">HOMSAP</span> <span class=\"o\">=</span> <span class=\"mi\">9606</span>\n",
       "<span class=\"n\">DANRER</span> <span class=\"o\">=</span> <span class=\"mi\">7955</span>\n",
       "<span class=\"n\">MYXINIDAE</span> <span class=\"o\">=</span> <span class=\"mi\">7762</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[30]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">mg</span> <span class=\"o\">=</span> <span class=\"n\">mygene</span><span class=\"o\">.</span><span class=\"n\">MyGeneInfo</span><span class=\"p\">()</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<p>Fix the uniprot names for querying</p>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[32]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">ortho_df</span><span class=\"p\">[</span><span class=\"s\">&#39;lamp10.fasta.x.Myx.pep.all.fa.db.tsv&#39;</span><span class=\"p\">]</span> <span class=\"o\">=</span> <span class=\"n\">ortho_df</span><span class=\"p\">[</span><span class=\"s\">&#39;lamp10.fasta.x.Myx.pep.all.fa.db.tsv&#39;</span><span class=\"p\">]</span><span class=\"o\">.</span><span class=\"n\">apply</span><span class=\"p\">(</span><span class=\"n\">uniprot_str</span><span class=\"p\">)</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[33]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">ortho_df</span><span class=\"p\">[</span><span class=\"s\">&#39;lamp10.fasta.x.braFlo.pep.all.fa.db.tsv&#39;</span><span class=\"p\">]</span> <span class=\"o\">=</span> <span class=\"n\">ortho_df</span><span class=\"p\">[</span><span class=\"s\">&#39;lamp10.fasta.x.braFlo.pep.all.fa.db.tsv&#39;</span><span class=\"p\">]</span><span class=\"o\">.</span><span class=\"n\">apply</span><span class=\"p\">(</span><span class=\"n\">uniprot_str</span><span class=\"p\">)</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<p>Make a DataFrame of just the potentially novel transcripts' orthology names</p>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[34]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">novel_df</span> <span class=\"o\">=</span> <span class=\"n\">ortho_df</span><span class=\"p\">[</span><span class=\"n\">has_ortho</span> <span class=\"o\">&amp;</span> <span class=\"n\">lamp_filter</span><span class=\"p\">]</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[138]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"k\">def</span> <span class=\"nf\">classify_orthology</span><span class=\"p\">(</span><span class=\"n\">df</span><span class=\"p\">):</span>\n",
       "    <span class=\"sd\">&#39;&#39;&#39;</span>\n",
       "<span class=\"sd\">    Given a DataFrame containing the names of transcript orthologies (either uniprot or ensembl identifiers),</span>\n",
       "<span class=\"sd\">    determine whether each orthology corresponds to a gene in the cyclostomata, cephalochordata, or </span>\n",
       "<span class=\"sd\">    gnathostomata lineages. Returns a list of tuples of booleans with the results, of the form:</span>\n",
       "<span class=\"sd\">    </span>\n",
       "<span class=\"sd\">    [(in cephalochordata, in gnathostomata, in cyclostomata),</span>\n",
       "<span class=\"sd\">     (...),</span>\n",
       "<span class=\"sd\">     ...]</span>\n",
       "<span class=\"sd\">     </span>\n",
       "<span class=\"sd\">     WARNING: This uses the mygene API and makes many complex queries; accordingly, it&#39;s slow as hell</span>\n",
       "<span class=\"sd\">    &#39;&#39;&#39;</span>\n",
       "    <span class=\"n\">species</span><span class=\"o\">=</span><span class=\"s\">&#39;,&#39;</span><span class=\"o\">.</span><span class=\"n\">join</span><span class=\"p\">(</span><span class=\"nb\">str</span><span class=\"p\">(</span><span class=\"n\">s</span><span class=\"p\">)</span> <span class=\"k\">for</span> <span class=\"n\">s</span> <span class=\"ow\">in</span> <span class=\"p\">[</span><span class=\"n\">BRAFLO</span><span class=\"p\">,</span> <span class=\"n\">MUSMUS</span><span class=\"p\">,</span> <span class=\"n\">HOMSAP</span><span class=\"p\">,</span> <span class=\"n\">DANRER</span><span class=\"p\">,</span> <span class=\"n\">MYXINIDAE</span><span class=\"p\">])</span>\n",
       "    <span class=\"n\">classifications</span> <span class=\"o\">=</span> <span class=\"p\">[]</span>\n",
       "    <span class=\"k\">for</span> <span class=\"n\">n</span><span class=\"p\">,</span> <span class=\"p\">(</span><span class=\"n\">tr_id</span><span class=\"p\">,</span> <span class=\"n\">row</span><span class=\"p\">)</span> <span class=\"ow\">in</span> <span class=\"nb\">enumerate</span><span class=\"p\">(</span><span class=\"n\">df</span><span class=\"o\">.</span><span class=\"n\">iterrows</span><span class=\"p\">()):</span>\n",
       "        <span class=\"k\">if</span> <span class=\"n\">n</span> <span class=\"o\">%</span> <span class=\"mi\">1000</span> <span class=\"o\">==</span> <span class=\"mi\">0</span><span class=\"p\">:</span>\n",
       "            <span class=\"k\">print</span> <span class=\"s\">&#39;Classified {n}...&#39;</span><span class=\"o\">.</span><span class=\"n\">format</span><span class=\"p\">(</span><span class=\"n\">n</span><span class=\"o\">=</span><span class=\"n\">n</span><span class=\"p\">)</span>\n",
       "        <span class=\"n\">queries</span> <span class=\"o\">=</span> <span class=\"p\">[</span><span class=\"n\">q</span> <span class=\"k\">for</span> <span class=\"n\">q</span> <span class=\"ow\">in</span> <span class=\"nb\">list</span><span class=\"p\">(</span><span class=\"n\">row</span><span class=\"p\">)</span> <span class=\"k\">if</span> <span class=\"nb\">type</span><span class=\"p\">(</span><span class=\"n\">q</span><span class=\"p\">)</span> <span class=\"ow\">is</span> <span class=\"nb\">str</span><span class=\"p\">]</span>\n",
       "        <span class=\"n\">results</span> <span class=\"o\">=</span> <span class=\"n\">mg</span><span class=\"o\">.</span><span class=\"n\">querymany</span><span class=\"p\">(</span><span class=\"n\">queries</span><span class=\"p\">,</span> <span class=\"n\">scopes</span><span class=\"o\">=</span><span class=\"s\">&#39;uniprot,ensemblprotein,ensembltranscript&#39;</span><span class=\"p\">,</span> \n",
       "                               <span class=\"n\">species</span><span class=\"o\">=</span><span class=\"n\">species</span><span class=\"p\">,</span> <span class=\"n\">as_dataframe</span><span class=\"o\">=</span><span class=\"bp\">True</span><span class=\"p\">,</span> <span class=\"n\">verbose</span><span class=\"o\">=</span><span class=\"bp\">False</span><span class=\"p\">)</span>\n",
       "        <span class=\"k\">try</span><span class=\"p\">:</span>\n",
       "            <span class=\"c\"># remove failed queries, if there are any</span>\n",
       "            <span class=\"n\">results</span> <span class=\"o\">=</span> <span class=\"n\">results</span><span class=\"p\">[</span><span class=\"n\">results</span><span class=\"p\">[</span><span class=\"s\">&#39;notfound&#39;</span><span class=\"p\">]</span> <span class=\"o\">!=</span> <span class=\"bp\">True</span><span class=\"p\">]</span>\n",
       "        <span class=\"k\">except</span> <span class=\"ne\">KeyError</span><span class=\"p\">:</span>\n",
       "            <span class=\"k\">pass</span>\n",
       "        \n",
       "        <span class=\"k\">def</span> <span class=\"nf\">check_results</span><span class=\"p\">(</span><span class=\"n\">res_list</span><span class=\"p\">):</span>\n",
       "            <span class=\"k\">for</span> <span class=\"n\">res</span> <span class=\"ow\">in</span> <span class=\"n\">res_list</span><span class=\"p\">:</span>\n",
       "                <span class=\"k\">if</span> <span class=\"s\">&#39;notfound&#39;</span> <span class=\"ow\">not</span> <span class=\"ow\">in</span> <span class=\"n\">res</span><span class=\"p\">:</span>\n",
       "                    <span class=\"k\">return</span> <span class=\"bp\">True</span>\n",
       "            <span class=\"k\">return</span> <span class=\"bp\">False</span>\n",
       "        \n",
       "        <span class=\"c\"># Check if any are in the cephalochorate lineage</span>\n",
       "        <span class=\"k\">try</span><span class=\"p\">:</span>\n",
       "            <span class=\"n\">ceph_results</span> <span class=\"o\">=</span> <span class=\"n\">mg</span><span class=\"o\">.</span><span class=\"n\">querymany</span><span class=\"p\">(</span><span class=\"nb\">list</span><span class=\"p\">(</span><span class=\"n\">results</span><span class=\"o\">.</span><span class=\"n\">symbol</span><span class=\"p\">),</span> <span class=\"n\">scopes</span><span class=\"o\">=</span><span class=\"s\">&#39;symbol&#39;</span><span class=\"p\">,</span> <span class=\"n\">fields</span><span class=\"o\">=</span><span class=\"s\">&#39;taxid&#39;</span><span class=\"p\">,</span> \n",
       "                                    <span class=\"n\">species</span><span class=\"o\">=</span><span class=\"n\">CEPHALOCHORDATA</span><span class=\"p\">,</span> <span class=\"n\">include_tax_tree</span><span class=\"o\">=</span><span class=\"bp\">True</span><span class=\"p\">,</span> <span class=\"n\">verbose</span><span class=\"o\">=</span><span class=\"bp\">False</span><span class=\"p\">)</span>\n",
       "            <span class=\"n\">ceph</span> <span class=\"o\">=</span> <span class=\"n\">check_results</span><span class=\"p\">(</span><span class=\"n\">ceph_results</span><span class=\"p\">)</span>\n",
       "        <span class=\"k\">except</span> <span class=\"ne\">AttributeError</span><span class=\"p\">:</span>\n",
       "            <span class=\"n\">ceph</span> <span class=\"o\">=</span> <span class=\"bp\">False</span>\n",
       "        \n",
       "        <span class=\"c\"># Check cyclostomata lineage</span>\n",
       "        <span class=\"k\">try</span><span class=\"p\">:</span>\n",
       "            <span class=\"n\">cycl_results</span> <span class=\"o\">=</span> <span class=\"n\">mg</span><span class=\"o\">.</span><span class=\"n\">querymany</span><span class=\"p\">(</span><span class=\"nb\">list</span><span class=\"p\">(</span><span class=\"n\">results</span><span class=\"o\">.</span><span class=\"n\">symbol</span><span class=\"p\">),</span> <span class=\"n\">scopes</span><span class=\"o\">=</span><span class=\"s\">&#39;symbol&#39;</span><span class=\"p\">,</span> <span class=\"n\">fields</span><span class=\"o\">=</span><span class=\"s\">&#39;taxid&#39;</span><span class=\"p\">,</span> \n",
       "                                    <span class=\"n\">species</span><span class=\"o\">=</span><span class=\"n\">CYCLOSTOMATA</span><span class=\"p\">,</span> <span class=\"n\">include_tax_tree</span><span class=\"o\">=</span><span class=\"bp\">True</span><span class=\"p\">,</span> <span class=\"n\">verbose</span><span class=\"o\">=</span><span class=\"bp\">False</span><span class=\"p\">)</span>\n",
       "            <span class=\"n\">cycl</span> <span class=\"o\">=</span> <span class=\"n\">check_results</span><span class=\"p\">(</span><span class=\"n\">cycl_results</span><span class=\"p\">)</span>\n",
       "        <span class=\"k\">except</span> <span class=\"ne\">AttributeError</span><span class=\"p\">:</span>\n",
       "            <span class=\"n\">cycl</span> <span class=\"o\">=</span> <span class=\"bp\">False</span>\n",
       "        \n",
       "        <span class=\"c\"># Check gnathostomata lineage</span>\n",
       "        <span class=\"k\">try</span><span class=\"p\">:</span>\n",
       "            <span class=\"n\">gnat_results</span> <span class=\"o\">=</span> <span class=\"n\">mg</span><span class=\"o\">.</span><span class=\"n\">querymany</span><span class=\"p\">(</span><span class=\"nb\">list</span><span class=\"p\">(</span><span class=\"n\">results</span><span class=\"o\">.</span><span class=\"n\">symbol</span><span class=\"p\">),</span> <span class=\"n\">scopes</span><span class=\"o\">=</span><span class=\"s\">&#39;symbol&#39;</span><span class=\"p\">,</span> <span class=\"n\">fields</span><span class=\"o\">=</span><span class=\"s\">&#39;taxid&#39;</span><span class=\"p\">,</span> \n",
       "                                    <span class=\"n\">species</span><span class=\"o\">=</span><span class=\"n\">GNATHOSTOMATA</span><span class=\"p\">,</span> <span class=\"n\">include_tax_tree</span><span class=\"o\">=</span><span class=\"bp\">True</span><span class=\"p\">,</span> <span class=\"n\">verbose</span><span class=\"o\">=</span><span class=\"bp\">False</span><span class=\"p\">)</span>\n",
       "            <span class=\"n\">gnat</span><span class=\"o\">=</span> <span class=\"n\">check_results</span><span class=\"p\">(</span><span class=\"n\">gnat_results</span><span class=\"p\">)</span>\n",
       "        <span class=\"k\">except</span> <span class=\"ne\">AttributeError</span><span class=\"p\">:</span>\n",
       "            <span class=\"n\">gnat</span> <span class=\"o\">=</span> <span class=\"bp\">False</span>\n",
       "        \n",
       "        <span class=\"n\">classifications</span><span class=\"o\">.</span><span class=\"n\">append</span><span class=\"p\">((</span><span class=\"n\">ceph</span><span class=\"p\">,</span> <span class=\"n\">gnat</span><span class=\"p\">,</span> <span class=\"n\">cycl</span><span class=\"p\">))</span>\n",
       "    <span class=\"k\">return</span> <span class=\"n\">classifications</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[130]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">ortho_classes</span> <span class=\"o\">=</span> <span class=\"n\">classify_orthology</span><span class=\"p\">(</span><span class=\"n\">novel_df</span><span class=\"p\">)</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt\"></div>\n",
       "<div class=\"output_subarea output_stream output_stdout output_text\">\n",
       "<pre>Classified 0...\n",
       "Classified 1000...\n",
       "</pre>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[136]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">class_df</span> <span class=\"o\">=</span> <span class=\"n\">pd</span><span class=\"o\">.</span><span class=\"n\">DataFrame</span><span class=\"p\">(</span><span class=\"n\">ortho_classes</span><span class=\"p\">,</span> <span class=\"n\">columns</span><span class=\"o\">=</span><span class=\"p\">[</span><span class=\"s\">&#39;cephalochordata&#39;</span><span class=\"p\">,</span> <span class=\"s\">&#39;gnathostomata&#39;</span><span class=\"p\">,</span> <span class=\"s\">&#39;cyclostomata&#39;</span><span class=\"p\">],</span>\n",
       "                        <span class=\"n\">index</span><span class=\"o\">=</span><span class=\"n\">novel_df</span><span class=\"o\">.</span><span class=\"n\">index</span><span class=\"p\">)</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[139]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">class_df</span><span class=\"o\">.</span><span class=\"n\">head</span><span class=\"p\">()</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt output_prompt\">Out[139]:</div>\n",
       "\n",
       "<div class=\"output_html rendered_html output_subarea output_execute_result\">\n",
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cephalochordata</th>\n",
       "      <th>gnathostomata</th>\n",
       "      <th>cyclostomata</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>target_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>c366614_g4_i7</th>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>c329845_g1_i1</th>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>c363162_g3_i1</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>c351306_g1_i1</th>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>c351306_g1_i2</th>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[144]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">class_df</span> <span class=\"o\">=</span> <span class=\"n\">pd</span><span class=\"o\">.</span><span class=\"n\">merge</span><span class=\"p\">(</span><span class=\"n\">novel_df</span><span class=\"p\">,</span> <span class=\"n\">class_df</span><span class=\"p\">,</span> <span class=\"n\">left_index</span><span class=\"o\">=</span><span class=\"bp\">True</span><span class=\"p\">,</span> <span class=\"n\">right_index</span><span class=\"o\">=</span><span class=\"bp\">True</span><span class=\"p\">)</span>\n",
       "<span class=\"k\">del</span> <span class=\"n\">class_df</span><span class=\"p\">[</span><span class=\"s\">&#39;lamp10.fasta.x.petMar2.pep.fa.db.tsv&#39;</span><span class=\"p\">]</span>\n",
       "<span class=\"k\">del</span> <span class=\"n\">class_df</span><span class=\"p\">[</span><span class=\"s\">&#39;lamp10.fasta.x.petMar2.ncrna.fa.db.tsv&#39;</span><span class=\"p\">]</span>\n",
       "<span class=\"k\">del</span> <span class=\"n\">class_df</span><span class=\"p\">[</span><span class=\"s\">&#39;lamp10.fasta.x.petMar2.cds.fa.db.tsv&#39;</span><span class=\"p\">]</span>\n",
       "<span class=\"k\">del</span> <span class=\"n\">class_df</span><span class=\"p\">[</span><span class=\"s\">&#39;lamp10.fasta.x.petMar2.cdna.fa.db.tsv&#39;</span><span class=\"p\">]</span>\n",
       "<span class=\"k\">del</span> <span class=\"n\">class_df</span><span class=\"p\">[</span><span class=\"s\">&#39;lamp10.fasta.x.petMar2.fa.db.tsv&#39;</span><span class=\"p\">]</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<p>Orthologies specific to the gnathostomata or cyclostomata line -- they're in either gnathostomata or cyclostomata, but not in cephalochordata.</p>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[151]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"p\">(((</span><span class=\"n\">class_df</span><span class=\"o\">.</span><span class=\"n\">gnathostomata</span> <span class=\"o\">==</span> <span class=\"bp\">True</span><span class=\"p\">)</span> <span class=\"o\">|</span> <span class=\"p\">(</span><span class=\"n\">class_df</span><span class=\"o\">.</span><span class=\"n\">cyclostomata</span> <span class=\"o\">==</span> <span class=\"bp\">True</span><span class=\"p\">))</span> <span class=\"o\">&amp;</span> <span class=\"p\">(</span><span class=\"n\">class_df</span><span class=\"o\">.</span><span class=\"n\">cephalochordata</span> <span class=\"o\">==</span> <span class=\"bp\">False</span><span class=\"p\">))</span><span class=\"o\">.</span><span class=\"n\">sum</span><span class=\"p\">()</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt output_prompt\">Out[151]:</div>\n",
       "\n",
       "\n",
       "<div class=\"output_text output_subarea output_execute_result\">\n",
       "<pre>796</pre>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[162]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"p\">((</span><span class=\"n\">class_df</span><span class=\"o\">.</span><span class=\"n\">gnathostomata</span> <span class=\"o\">==</span> <span class=\"bp\">True</span><span class=\"p\">)</span> <span class=\"o\">&amp;</span> <span class=\"p\">(</span><span class=\"n\">class_df</span><span class=\"o\">.</span><span class=\"n\">cephalochordata</span> <span class=\"o\">==</span> <span class=\"bp\">False</span><span class=\"p\">))</span><span class=\"o\">.</span><span class=\"n\">sum</span><span class=\"p\">()</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt output_prompt\">Out[162]:</div>\n",
       "\n",
       "\n",
       "<div class=\"output_text output_subarea output_execute_result\">\n",
       "<pre>796</pre>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<p>It looks like it makes no difference whether we include cyclostomata. That is, we actually have up to 796 orthologies which are specific to jawed vertebrates but probably not a part of any existing cyclostome database. Cool! Given that, let's check what happens when we enforce both -- that is, genes which are specific to the vertebrate line.</p>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[165]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"p\">((</span><span class=\"n\">class_df</span><span class=\"o\">.</span><span class=\"n\">cyclostomata</span> <span class=\"o\">==</span> <span class=\"bp\">True</span><span class=\"p\">)</span> <span class=\"o\">&amp;</span> <span class=\"p\">(</span><span class=\"n\">class_df</span><span class=\"o\">.</span><span class=\"n\">gnathostomata</span> <span class=\"o\">==</span> <span class=\"bp\">True</span><span class=\"p\">)</span> <span class=\"o\">&amp;</span> <span class=\"p\">(</span><span class=\"n\">class_df</span><span class=\"o\">.</span><span class=\"n\">cephalochordata</span> <span class=\"o\">==</span> <span class=\"bp\">False</span><span class=\"p\">))</span><span class=\"o\">.</span><span class=\"n\">sum</span><span class=\"p\">()</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt output_prompt\">Out[165]:</div>\n",
       "\n",
       "\n",
       "<div class=\"output_text output_subarea output_execute_result\">\n",
       "<pre>60</pre>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<p>So 60 of those orthologies have cyclostomata support -- they likely aren't in sea lamprey resources, so they must either be from hagfish or another lamprey. How about when we exclude the entire cyclostomata lineage?</p>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[166]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"p\">((</span><span class=\"n\">class_df</span><span class=\"o\">.</span><span class=\"n\">cyclostomata</span> <span class=\"o\">==</span> <span class=\"bp\">False</span><span class=\"p\">)</span> <span class=\"o\">&amp;</span> <span class=\"p\">(</span><span class=\"n\">class_df</span><span class=\"o\">.</span><span class=\"n\">gnathostomata</span> <span class=\"o\">==</span> <span class=\"bp\">True</span><span class=\"p\">)</span> <span class=\"o\">&amp;</span> <span class=\"p\">(</span><span class=\"n\">class_df</span><span class=\"o\">.</span><span class=\"n\">cephalochordata</span> <span class=\"o\">==</span> <span class=\"bp\">False</span><span class=\"p\">))</span><span class=\"o\">.</span><span class=\"n\">sum</span><span class=\"p\">()</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt output_prompt\">Out[166]:</div>\n",
       "\n",
       "\n",
       "<div class=\"output_text output_subarea output_execute_result\">\n",
       "<pre>736</pre>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<p>Here's what we're really trying to get at -- that's 736 orthologies which are specific to the jawed vertebrate line according to current resources, but clearly have pretty strong support in our transcriptome. Alright, what are these genes?</p>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[170]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">novel_df</span><span class=\"p\">[((</span><span class=\"n\">class_df</span><span class=\"o\">.</span><span class=\"n\">cyclostomata</span> <span class=\"o\">==</span> <span class=\"bp\">False</span><span class=\"p\">)</span> <span class=\"o\">&amp;</span> <span class=\"p\">(</span><span class=\"n\">class_df</span><span class=\"o\">.</span><span class=\"n\">gnathostomata</span> <span class=\"o\">==</span> <span class=\"bp\">True</span><span class=\"p\">)</span> <span class=\"o\">&amp;</span> <span class=\"p\">(</span><span class=\"n\">class_df</span><span class=\"o\">.</span><span class=\"n\">cephalochordata</span> <span class=\"o\">==</span> <span class=\"bp\">False</span><span class=\"p\">))]</span><span class=\"o\">.</span><span class=\"n\">head</span><span class=\"p\">()</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt output_prompt\">Out[170]:</div>\n",
       "\n",
       "<div class=\"output_html rendered_html output_subarea output_execute_result\">\n",
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>lamp10.fasta.x.Myx.pep.all.fa.db.tsv</th>\n",
       "      <th>lamp10.fasta.x.braFlo.pep.all.fa.db.tsv</th>\n",
       "      <th>lamp10.fasta.x.danRer.pep.fa.db.tsv</th>\n",
       "      <th>lamp10.fasta.x.homSap.pep.fa.db.tsv</th>\n",
       "      <th>lamp10.fasta.x.musMus.pep.fa.db.tsv</th>\n",
       "      <th>lamp10.fasta.x.petMar2.fa.db.tsv</th>\n",
       "      <th>lamp10.fasta.x.petMar2.cdna.fa.db.tsv</th>\n",
       "      <th>lamp10.fasta.x.petMar2.cds.fa.db.tsv</th>\n",
       "      <th>lamp10.fasta.x.petMar2.ncrna.fa.db.tsv</th>\n",
       "      <th>lamp10.fasta.x.petMar2.pep.fa.db.tsv</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>target_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>c329845_g1_i1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSDARP00000105046</td>\n",
       "      <td>ENSP00000329735</td>\n",
       "      <td>ENSMUSP00000040433</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>c351306_g1_i1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSMUSP00000136481</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>c292244_g1_i1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSDARP00000018388</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>c360287_g1_i1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSDARP00000009179</td>\n",
       "      <td>ENSP00000256997</td>\n",
       "      <td>ENSMUSP00000108209</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>c265851_g1_i1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSP00000443978</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[190]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"k\">def</span> <span class=\"nf\">get_symbols</span><span class=\"p\">(</span><span class=\"n\">df</span><span class=\"p\">):</span>\n",
       "    <span class=\"n\">species</span><span class=\"o\">=</span><span class=\"s\">&#39;,&#39;</span><span class=\"o\">.</span><span class=\"n\">join</span><span class=\"p\">(</span><span class=\"nb\">str</span><span class=\"p\">(</span><span class=\"n\">s</span><span class=\"p\">)</span> <span class=\"k\">for</span> <span class=\"n\">s</span> <span class=\"ow\">in</span> <span class=\"p\">[</span><span class=\"n\">BRAFLO</span><span class=\"p\">,</span> <span class=\"n\">MUSMUS</span><span class=\"p\">,</span> <span class=\"n\">HOMSAP</span><span class=\"p\">,</span> <span class=\"n\">DANRER</span><span class=\"p\">,</span> <span class=\"n\">MYXINIDAE</span><span class=\"p\">])</span>\n",
       "    <span class=\"n\">queries</span> <span class=\"o\">=</span> <span class=\"p\">[</span><span class=\"n\">q</span> <span class=\"k\">for</span> <span class=\"n\">q</span> <span class=\"ow\">in</span> <span class=\"nb\">set</span><span class=\"p\">(</span><span class=\"n\">df</span><span class=\"o\">.</span><span class=\"n\">values</span><span class=\"o\">.</span><span class=\"n\">flatten</span><span class=\"p\">())</span> <span class=\"k\">if</span> <span class=\"nb\">type</span><span class=\"p\">(</span><span class=\"n\">q</span><span class=\"p\">)</span> <span class=\"ow\">is</span> <span class=\"nb\">str</span><span class=\"p\">]</span>\n",
       "    <span class=\"n\">results</span> <span class=\"o\">=</span> <span class=\"n\">mg</span><span class=\"o\">.</span><span class=\"n\">querymany</span><span class=\"p\">(</span><span class=\"n\">queries</span><span class=\"p\">,</span> <span class=\"n\">scopes</span><span class=\"o\">=</span><span class=\"s\">&#39;uniprot,ensemblprotein,ensembltranscript&#39;</span><span class=\"p\">,</span> \n",
       "                               <span class=\"n\">species</span><span class=\"o\">=</span><span class=\"n\">species</span><span class=\"p\">,</span> <span class=\"n\">as_dataframe</span><span class=\"o\">=</span><span class=\"bp\">True</span><span class=\"p\">,</span> <span class=\"n\">verbose</span><span class=\"o\">=</span><span class=\"bp\">False</span><span class=\"p\">)</span>\n",
       "    <span class=\"n\">symbols</span> <span class=\"o\">=</span> <span class=\"nb\">set</span><span class=\"p\">([</span><span class=\"n\">r</span><span class=\"o\">.</span><span class=\"n\">upper</span><span class=\"p\">()</span> <span class=\"k\">for</span> <span class=\"n\">r</span> <span class=\"ow\">in</span> <span class=\"n\">results</span><span class=\"o\">.</span><span class=\"n\">symbol</span> <span class=\"k\">if</span> <span class=\"nb\">type</span><span class=\"p\">(</span><span class=\"n\">r</span><span class=\"p\">)</span> <span class=\"ow\">in</span> <span class=\"p\">[</span><span class=\"nb\">str</span><span class=\"p\">,</span><span class=\"nb\">unicode</span><span class=\"p\">]])</span>\n",
       "    <span class=\"k\">return</span> <span class=\"n\">symbols</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[191]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">jvert_specific</span> <span class=\"o\">=</span> <span class=\"n\">get_symbols</span><span class=\"p\">(</span><span class=\"n\">novel_df</span><span class=\"p\">[((</span><span class=\"n\">class_df</span><span class=\"o\">.</span><span class=\"n\">cyclostomata</span> <span class=\"o\">==</span> <span class=\"bp\">False</span><span class=\"p\">)</span> <span class=\"o\">&amp;</span> <span class=\"p\">(</span><span class=\"n\">class_df</span><span class=\"o\">.</span><span class=\"n\">gnathostomata</span> <span class=\"o\">==</span> <span class=\"bp\">True</span><span class=\"p\">)</span> <span class=\"o\">&amp;</span> \n",
       "                                       <span class=\"p\">(</span><span class=\"n\">class_df</span><span class=\"o\">.</span><span class=\"n\">cephalochordata</span> <span class=\"o\">==</span> <span class=\"bp\">False</span><span class=\"p\">))])</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[193]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"nb\">len</span><span class=\"p\">(</span><span class=\"n\">jvert_specific</span><span class=\"p\">)</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt output_prompt\">Out[193]:</div>\n",
       "\n",
       "\n",
       "<div class=\"output_text output_subarea output_execute_result\">\n",
       "<pre>780</pre>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[197]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"p\">(</span><span class=\"n\">class_df</span><span class=\"o\">.</span><span class=\"n\">sum</span><span class=\"p\">(</span><span class=\"n\">axis</span><span class=\"o\">=</span><span class=\"mi\">1</span><span class=\"p\">)</span> <span class=\"o\">==</span> <span class=\"mi\">0</span><span class=\"p\">)</span><span class=\"o\">.</span><span class=\"n\">sum</span><span class=\"p\">()</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt output_prompt\">Out[197]:</div>\n",
       "\n",
       "\n",
       "<div class=\"output_text output_subarea output_execute_result\">\n",
       "<pre>103</pre>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<h2 id=\"Germline-DNA-Samples\">Germline DNA Samples<a class=\"anchor-link\" href=\"#Germline-DNA-Samples\">&#182;</a></h2>\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<p>We've got some nice shiny new sperm DNA samples - let's play with them some. First things first, we'll check out their FastQC results to make sure they're not crap.</p>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[30]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">fastqc_folders</span> <span class=\"o\">=</span> <span class=\"nb\">sorted</span><span class=\"p\">(</span><span class=\"n\">glob</span><span class=\"o\">.</span><span class=\"n\">glob</span><span class=\"p\">(</span><span class=\"s\">&#39;*fastqc&#39;</span><span class=\"p\">))</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[31]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"o\">!</span>ls 2_ATCACG_L001_R1_001_fastqc/Images\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt\"></div>\n",
       "<div class=\"output_subarea output_stream output_stdout output_text\">\n",
       "<pre>duplication_levels.png\t per_base_sequence_content.png\r\n",
       "kmer_profiles.png\t per_sequence_gc_content.png\r\n",
       "per_base_gc_content.png  per_sequence_quality.png\r\n",
       "per_base_n_content.png\t sequence_length_distribution.png\r\n",
       "per_base_quality.png\r\n",
       "</pre>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[32]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"c\"># quick utility function to generate a table of images</span>\n",
       "<span class=\"k\">def</span> <span class=\"nf\">get_img_table</span><span class=\"p\">(</span><span class=\"n\">folders</span><span class=\"p\">,</span> <span class=\"n\">image</span><span class=\"p\">):</span>\n",
       "    <span class=\"n\">htmlout</span> <span class=\"o\">=</span> <span class=\"s\">&#39;&lt;table&gt;&#39;</span>\n",
       "    <span class=\"k\">for</span> <span class=\"n\">i</span> <span class=\"ow\">in</span> <span class=\"nb\">xrange</span><span class=\"p\">(</span><span class=\"mi\">0</span><span class=\"p\">,</span> <span class=\"nb\">len</span><span class=\"p\">(</span><span class=\"n\">folders</span><span class=\"p\">),</span> <span class=\"mi\">2</span><span class=\"p\">):</span>\n",
       "        <span class=\"n\">left</span> <span class=\"o\">=</span> <span class=\"n\">fastqc_folders</span><span class=\"p\">[</span><span class=\"n\">i</span><span class=\"p\">]</span>\n",
       "        <span class=\"n\">right</span> <span class=\"o\">=</span> <span class=\"n\">fastqc_folders</span><span class=\"p\">[</span><span class=\"n\">i</span><span class=\"o\">+</span><span class=\"mi\">1</span><span class=\"p\">]</span>\n",
       "        <span class=\"n\">htmlout</span> <span class=\"o\">+=</span> <span class=\"s\">&#39;&lt;tr&gt;&#39;</span>\n",
       "        <span class=\"n\">htmlout</span> <span class=\"o\">+=</span> <span class=\"s\">&#39;&lt;td align=&quot;center&quot;&gt;&#39;</span> <span class=\"o\">+</span> <span class=\"n\">left</span> <span class=\"o\">+</span> <span class=\"s\">&#39;, LEFT&lt;/td&gt;&#39;</span>\n",
       "        <span class=\"n\">htmlout</span> <span class=\"o\">+=</span> <span class=\"s\">&#39;&lt;td align=&quot;center&quot;&gt;&#39;</span> <span class=\"o\">+</span> <span class=\"n\">right</span> <span class=\"o\">+</span> <span class=\"s\">&#39;, RIGHT&lt;/td&gt;&#39;</span>\n",
       "        <span class=\"n\">htmlout</span> <span class=\"o\">+=</span> <span class=\"s\">&#39;&lt;/td&gt;&#39;</span>\n",
       "        \n",
       "        <span class=\"n\">htmlout</span> <span class=\"o\">+=</span> <span class=\"s\">&#39;&lt;tr&gt;&#39;</span>\n",
       "        <span class=\"n\">htmlout</span> <span class=\"o\">+=</span> <span class=\"s\">&#39;&lt;td&gt;&lt;img src=&quot;&#39;</span> <span class=\"o\">+</span> <span class=\"n\">os</span><span class=\"o\">.</span><span class=\"n\">path</span><span class=\"o\">.</span><span class=\"n\">join</span><span class=\"p\">(</span><span class=\"n\">left</span><span class=\"p\">,</span> <span class=\"s\">&#39;Images&#39;</span><span class=\"p\">,</span> <span class=\"n\">image</span><span class=\"p\">)</span> <span class=\"o\">+</span> <span class=\"s\">&#39;&quot;/&gt;&lt;/td&gt;&#39;</span>\n",
       "        <span class=\"n\">htmlout</span> <span class=\"o\">+=</span> <span class=\"s\">&#39;&lt;td&gt;&lt;img src=&quot;&#39;</span> <span class=\"o\">+</span> <span class=\"n\">os</span><span class=\"o\">.</span><span class=\"n\">path</span><span class=\"o\">.</span><span class=\"n\">join</span><span class=\"p\">(</span><span class=\"n\">right</span><span class=\"p\">,</span> <span class=\"s\">&#39;Images&#39;</span><span class=\"p\">,</span> <span class=\"n\">image</span><span class=\"p\">)</span> <span class=\"o\">+</span> <span class=\"s\">&#39;&quot;/&gt;&lt;/td&gt;&#39;</span>\n",
       "        <span class=\"n\">htmlout</span> <span class=\"o\">+=</span> <span class=\"s\">&#39;&lt;/tr&gt;&#39;</span>\n",
       "    <span class=\"n\">htmlout</span> <span class=\"o\">+=</span> <span class=\"s\">&#39;&lt;/table&gt;&#39;</span>\n",
       "    <span class=\"k\">return</span> <span class=\"n\">htmlout</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<h4 id=\"Per-Base-Sequence-Quality\">Per-Base Sequence Quality<a class=\"anchor-link\" href=\"#Per-Base-Sequence-Quality\">&#182;</a></h4><p>These look pretty normal -- in fact, the right fragments are a little better than I might expect for many runs.</p>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[33]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">display</span><span class=\"p\">(</span><span class=\"n\">HTML</span><span class=\"p\">(</span><span class=\"n\">get_img_table</span><span class=\"p\">(</span><span class=\"n\">fastqc_folders</span><span class=\"p\">,</span> <span class=\"s\">&#39;per_base_quality.png&#39;</span><span class=\"p\">)))</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt\"></div>\n",
       "\n",
       "<div class=\"output_html rendered_html output_subarea \">\n",
       "<table><tr><td align=\"center\">2_ATCACG_L001_R1_001_fastqc, LEFT</td><td align=\"center\">2_ATCACG_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"2_ATCACG_L001_R1_001_fastqc/Images/per_base_quality.png\"/></td><td><img src=\"2_ATCACG_L001_R2_001_fastqc/Images/per_base_quality.png\"/></td></tr><tr><td align=\"center\">2_TGACCA_L001_R1_001_fastqc, LEFT</td><td align=\"center\">2_TGACCA_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"2_TGACCA_L001_R1_001_fastqc/Images/per_base_quality.png\"/></td><td><img src=\"2_TGACCA_L001_R2_001_fastqc/Images/per_base_quality.png\"/></td></tr><tr><td align=\"center\">4_CCGTCC_L001_R1_001_fastqc, LEFT</td><td align=\"center\">4_CCGTCC_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"4_CCGTCC_L001_R1_001_fastqc/Images/per_base_quality.png\"/></td><td><img src=\"4_CCGTCC_L001_R2_001_fastqc/Images/per_base_quality.png\"/></td></tr><tr><td align=\"center\">4_GATCAG_L001_R1_001_fastqc, LEFT</td><td align=\"center\">4_GATCAG_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"4_GATCAG_L001_R1_001_fastqc/Images/per_base_quality.png\"/></td><td><img src=\"4_GATCAG_L001_R2_001_fastqc/Images/per_base_quality.png\"/></td></tr></table>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<h4 id=\"GC-Content\">GC Content<a class=\"anchor-link\" href=\"#GC-Content\">&#182;</a></h4>\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<p>For the most part, these results line up with what we expect from the lamprey genome paper: ~46% GC content for the whole genome (FastQC throws a warning for this check, but we can safely ignore it based on our prior knowledge). Curiously, in the genome paper, they report that coding regions had a GC content of ~61%, but we don't really see a bump in the distribution there -- instead, we see a bump at ~84% in all samples. There isn't anything unexpected in the per-<em>base</em> figures, so we can at least assume there isn't a technical artifact common to all reads.</p>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[34]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">display</span><span class=\"p\">(</span><span class=\"n\">HTML</span><span class=\"p\">(</span><span class=\"n\">get_img_table</span><span class=\"p\">(</span><span class=\"n\">fastqc_folders</span><span class=\"p\">,</span> <span class=\"s\">&#39;per_sequence_gc_content.png&#39;</span><span class=\"p\">)))</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt\"></div>\n",
       "\n",
       "<div class=\"output_html rendered_html output_subarea \">\n",
       "<table><tr><td align=\"center\">2_ATCACG_L001_R1_001_fastqc, LEFT</td><td align=\"center\">2_ATCACG_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"2_ATCACG_L001_R1_001_fastqc/Images/per_sequence_gc_content.png\"/></td><td><img src=\"2_ATCACG_L001_R2_001_fastqc/Images/per_sequence_gc_content.png\"/></td></tr><tr><td align=\"center\">2_TGACCA_L001_R1_001_fastqc, LEFT</td><td align=\"center\">2_TGACCA_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"2_TGACCA_L001_R1_001_fastqc/Images/per_sequence_gc_content.png\"/></td><td><img src=\"2_TGACCA_L001_R2_001_fastqc/Images/per_sequence_gc_content.png\"/></td></tr><tr><td align=\"center\">4_CCGTCC_L001_R1_001_fastqc, LEFT</td><td align=\"center\">4_CCGTCC_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"4_CCGTCC_L001_R1_001_fastqc/Images/per_sequence_gc_content.png\"/></td><td><img src=\"4_CCGTCC_L001_R2_001_fastqc/Images/per_sequence_gc_content.png\"/></td></tr><tr><td align=\"center\">4_GATCAG_L001_R1_001_fastqc, LEFT</td><td align=\"center\">4_GATCAG_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"4_GATCAG_L001_R1_001_fastqc/Images/per_sequence_gc_content.png\"/></td><td><img src=\"4_GATCAG_L001_R2_001_fastqc/Images/per_sequence_gc_content.png\"/></td></tr></table>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[35]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">display</span><span class=\"p\">(</span><span class=\"n\">HTML</span><span class=\"p\">(</span><span class=\"n\">get_img_table</span><span class=\"p\">(</span><span class=\"n\">fastqc_folders</span><span class=\"p\">,</span> <span class=\"s\">&#39;per_base_gc_content.png&#39;</span><span class=\"p\">)))</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt\"></div>\n",
       "\n",
       "<div class=\"output_html rendered_html output_subarea \">\n",
       "<table><tr><td align=\"center\">2_ATCACG_L001_R1_001_fastqc, LEFT</td><td align=\"center\">2_ATCACG_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"2_ATCACG_L001_R1_001_fastqc/Images/per_base_gc_content.png\"/></td><td><img src=\"2_ATCACG_L001_R2_001_fastqc/Images/per_base_gc_content.png\"/></td></tr><tr><td align=\"center\">2_TGACCA_L001_R1_001_fastqc, LEFT</td><td align=\"center\">2_TGACCA_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"2_TGACCA_L001_R1_001_fastqc/Images/per_base_gc_content.png\"/></td><td><img src=\"2_TGACCA_L001_R2_001_fastqc/Images/per_base_gc_content.png\"/></td></tr><tr><td align=\"center\">4_CCGTCC_L001_R1_001_fastqc, LEFT</td><td align=\"center\">4_CCGTCC_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"4_CCGTCC_L001_R1_001_fastqc/Images/per_base_gc_content.png\"/></td><td><img src=\"4_CCGTCC_L001_R2_001_fastqc/Images/per_base_gc_content.png\"/></td></tr><tr><td align=\"center\">4_GATCAG_L001_R1_001_fastqc, LEFT</td><td align=\"center\">4_GATCAG_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"4_GATCAG_L001_R1_001_fastqc/Images/per_base_gc_content.png\"/></td><td><img src=\"4_GATCAG_L001_R2_001_fastqc/Images/per_base_gc_content.png\"/></td></tr></table>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<p>Once more, nothing all too exciting -- a bit of A bias toward the end of the righ reads, which coincides with the expected drop in quality.</p>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[36]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">display</span><span class=\"p\">(</span><span class=\"n\">HTML</span><span class=\"p\">(</span><span class=\"n\">get_img_table</span><span class=\"p\">(</span><span class=\"n\">fastqc_folders</span><span class=\"p\">,</span> <span class=\"s\">&#39;per_base_sequence_content.png&#39;</span><span class=\"p\">)))</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt\"></div>\n",
       "\n",
       "<div class=\"output_html rendered_html output_subarea \">\n",
       "<table><tr><td align=\"center\">2_ATCACG_L001_R1_001_fastqc, LEFT</td><td align=\"center\">2_ATCACG_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"2_ATCACG_L001_R1_001_fastqc/Images/per_base_sequence_content.png\"/></td><td><img src=\"2_ATCACG_L001_R2_001_fastqc/Images/per_base_sequence_content.png\"/></td></tr><tr><td align=\"center\">2_TGACCA_L001_R1_001_fastqc, LEFT</td><td align=\"center\">2_TGACCA_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"2_TGACCA_L001_R1_001_fastqc/Images/per_base_sequence_content.png\"/></td><td><img src=\"2_TGACCA_L001_R2_001_fastqc/Images/per_base_sequence_content.png\"/></td></tr><tr><td align=\"center\">4_CCGTCC_L001_R1_001_fastqc, LEFT</td><td align=\"center\">4_CCGTCC_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"4_CCGTCC_L001_R1_001_fastqc/Images/per_base_sequence_content.png\"/></td><td><img src=\"4_CCGTCC_L001_R2_001_fastqc/Images/per_base_sequence_content.png\"/></td></tr><tr><td align=\"center\">4_GATCAG_L001_R1_001_fastqc, LEFT</td><td align=\"center\">4_GATCAG_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"4_GATCAG_L001_R1_001_fastqc/Images/per_base_sequence_content.png\"/></td><td><img src=\"4_GATCAG_L001_R2_001_fastqc/Images/per_base_sequence_content.png\"/></td></tr></table>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<p>Seems a little odd that we get this slow increase in G/C homopolymers over the length of our reads. Need to investigate further.</p>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[37]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">display</span><span class=\"p\">(</span><span class=\"n\">HTML</span><span class=\"p\">(</span><span class=\"n\">get_img_table</span><span class=\"p\">(</span><span class=\"n\">fastqc_folders</span><span class=\"p\">,</span> <span class=\"s\">&#39;kmer_profiles.png&#39;</span><span class=\"p\">)))</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt\"></div>\n",
       "\n",
       "<div class=\"output_html rendered_html output_subarea \">\n",
       "<table><tr><td align=\"center\">2_ATCACG_L001_R1_001_fastqc, LEFT</td><td align=\"center\">2_ATCACG_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"2_ATCACG_L001_R1_001_fastqc/Images/kmer_profiles.png\"/></td><td><img src=\"2_ATCACG_L001_R2_001_fastqc/Images/kmer_profiles.png\"/></td></tr><tr><td align=\"center\">2_TGACCA_L001_R1_001_fastqc, LEFT</td><td align=\"center\">2_TGACCA_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"2_TGACCA_L001_R1_001_fastqc/Images/kmer_profiles.png\"/></td><td><img src=\"2_TGACCA_L001_R2_001_fastqc/Images/kmer_profiles.png\"/></td></tr><tr><td align=\"center\">4_CCGTCC_L001_R1_001_fastqc, LEFT</td><td align=\"center\">4_CCGTCC_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"4_CCGTCC_L001_R1_001_fastqc/Images/kmer_profiles.png\"/></td><td><img src=\"4_CCGTCC_L001_R2_001_fastqc/Images/kmer_profiles.png\"/></td></tr><tr><td align=\"center\">4_GATCAG_L001_R1_001_fastqc, LEFT</td><td align=\"center\">4_GATCAG_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"4_GATCAG_L001_R1_001_fastqc/Images/kmer_profiles.png\"/></td><td><img src=\"4_GATCAG_L001_R2_001_fastqc/Images/kmer_profiles.png\"/></td></tr></table>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[38]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre><span class=\"n\">display</span><span class=\"p\">(</span><span class=\"n\">HTML</span><span class=\"p\">(</span><span class=\"n\">get_img_table</span><span class=\"p\">(</span><span class=\"n\">fastqc_folders</span><span class=\"p\">,</span> <span class=\"s\">&#39;duplication_levels.png&#39;</span><span class=\"p\">)))</span>\n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "<div class=\"output_wrapper\">\n",
       "<div class=\"output\">\n",
       "\n",
       "\n",
       "<div class=\"output_area\"><div class=\"prompt\"></div>\n",
       "\n",
       "<div class=\"output_html rendered_html output_subarea \">\n",
       "<table><tr><td align=\"center\">2_ATCACG_L001_R1_001_fastqc, LEFT</td><td align=\"center\">2_ATCACG_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"2_ATCACG_L001_R1_001_fastqc/Images/duplication_levels.png\"/></td><td><img src=\"2_ATCACG_L001_R2_001_fastqc/Images/duplication_levels.png\"/></td></tr><tr><td align=\"center\">2_TGACCA_L001_R1_001_fastqc, LEFT</td><td align=\"center\">2_TGACCA_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"2_TGACCA_L001_R1_001_fastqc/Images/duplication_levels.png\"/></td><td><img src=\"2_TGACCA_L001_R2_001_fastqc/Images/duplication_levels.png\"/></td></tr><tr><td align=\"center\">4_CCGTCC_L001_R1_001_fastqc, LEFT</td><td align=\"center\">4_CCGTCC_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"4_CCGTCC_L001_R1_001_fastqc/Images/duplication_levels.png\"/></td><td><img src=\"4_CCGTCC_L001_R2_001_fastqc/Images/duplication_levels.png\"/></td></tr><tr><td align=\"center\">4_GATCAG_L001_R1_001_fastqc, LEFT</td><td align=\"center\">4_GATCAG_L001_R2_001_fastqc, RIGHT</td></td><tr><td><img src=\"4_GATCAG_L001_R1_001_fastqc/Images/duplication_levels.png\"/></td><td><img src=\"4_GATCAG_L001_R2_001_fastqc/Images/duplication_levels.png\"/></td></tr></table>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing text_cell rendered\">\n",
       "<div class=\"prompt input_prompt\">\n",
       "</div>\n",
       "<div class=\"inner_cell\">\n",
       "<div class=\"text_cell_render border-box-sizing rendered_html\">\n",
       "<p>Summary: we should probably run a trimmer on these samples before using them. Shocker!</p>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "<div class=\"cell border-box-sizing code_cell rendered\">\n",
       "<div class=\"input\">\n",
       "<div class=\"prompt input_prompt\">In&nbsp;[&nbsp;]:</div>\n",
       "<div class=\"inner_cell\">\n",
       "    <div class=\"input_area\">\n",
       "<div class=\" highlight hl-ipython2\"><pre> \n",
       "</pre></div>\n",
       "\n",
       "</div>\n",
       "</div>\n",
       "</div>\n",
       "\n",
       "</div>\n",
       "    </div>\n",
       "  </div>\n",
       "\n",
       "\n",
       "    <div class=\"back-to-top hidden-print text-center\">\n",
       "      <a href=\"#\">Back to top</a>\n",
       "    </div>\n",
       "\n",
       "    \n",
       "      <footer class=\"footer hidden-print col-md-12\">\n",
       "        <div class=\"col-md-4\">\n",
       "          <p>\n",
       "            This web site does not host notebooks, it only renders notebooks\n",
       "            available on other websites.\n",
       "          </p>\n",
       "        </div>\n",
       "\n",
       "        <div class=\"col-md-4\">\n",
       "          <p>\n",
       "            Delivered by <a href=\"http://www.fastly.com/\">Fastly</a>, \n",
       "            Rendered by <a href=\"https://developer.rackspace.com/?nbviewer=awesome\">Rackspace</a>\n",
       "          </p>\n",
       "          <p>\n",
       "            nbviewer GitHub <a href=\"https://github.com/jupyter/nbviewer\">repository</a>.\n",
       "          </p>\n",
       "        </div>\n",
       "\n",
       "        <div class=\"col-md-4\">\n",
       "          \n",
       "  \n",
       "            \n",
       "              <p>\n",
       "                nbviewer version:\n",
       "                <a href=\"https://github.com/jupyter/nbviewer/commit/03d6df082a04afe885eb1a7dd6dcd25b45a89181\">\n",
       "                  03d6df0\n",
       "                </a>\n",
       "              </p>\n",
       "            \n",
       "          \n",
       "  \n",
       "  <p class=\"text-muted\">\n",
       "    IPython version: 4.0.0\n",
       "    \n",
       "  </p>\n",
       "  \n",
       "\n",
       "          \n",
       "  \n",
       "  \n",
       "  <p class=\"text-muted\">\n",
       "    Rendered\n",
       "    <span class='date' data-date='Tue, 15 Sep 2015 18:17:00 UTC' title='Tue, 15 Sep 2015 18:17:00 UTC'>(Tue, 15 Sep 2015 18:17:00 UTC)</span>\n",
       "  </p>\n",
       "  \n",
       "\n",
       "        </div>\n",
       "      </footer>\n",
       "    \n",
       "\n",
       "  </div>\n",
       "\n",
       "  <script src=\"/static/components/bootstrap/js/bootstrap.min.js\"></script>\n",
       "  <script src=\"/static/components/headroom.js/dist/headroom.min.js\"></script>\n",
       "  <script src=\"/static/components/headroom.js/dist/jQuery.headroom.min.js\"></script>\n",
       "\n",
       "  \n",
       "  \n",
       "  <script>\n",
       "    $(function(){ $(\"#menubar\").headroom({\n",
       "      tolerance: 5,\n",
       "      offset: 205,\n",
       "      classes: {\n",
       "        initial: \"animated\",\n",
       "        pinned: \"slideInDown\",\n",
       "        unpinned: \"slideOutUp\"\n",
       "      }\n",
       "    })});\n",
       "  </script>\n",
       "\n",
       "\n",
       "  <script>\n",
       "    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){\n",
       "      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),\n",
       "      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)\n",
       "    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');\n",
       "    ga('create', 'UA-38683231-2', 'auto');\n",
       "    ga('send', 'pageview');\n",
       "  </script>\n",
       "  <script>\n",
       "      var update_date  = function(){\n",
       "          var str = window.moment($('.date').data('date')).fromNow();\n",
       "          $('.date').text(str);\n",
       "      }\n",
       "      setInterval(update_date , 61*1000);\n",
       "      $(update_date);\n",
       "  </script>\n",
       "  <!--NEW RELIC Stop Perf Measurement-->\n",
       "  \n",
       "  <!--NEW RELIC End-->\n",
       "</body>\n",
       "</html>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "HTML('http://nbviewer.ipython.org/github/dib-lab/2015-petMarSB/blob/master/notebooks/petmar-genome-completeness.ipynb')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "* Purpose-built services with limited scope are more useful and useable than monolithic ones. mygene for programmatic exploration!\n",
    "    - http://mygene.info/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false,
    "scrolled": false,
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{u'hits': [{u'Vega': u'OTTDARG00000019822',\n",
       "   u'ZFIN': u'ZDB-GENE-990415-106',\n",
       "   u'_id': u'30341',\n",
       "   u'_score': 0.8095548,\n",
       "   u'_timestamp': u'2015-08-02T00:00:00',\n",
       "   u'accession': {u'genomic': [u'AL645782',\n",
       "     u'BX927395',\n",
       "     u'DU644552',\n",
       "     u'NC_007114',\n",
       "     u'NW_003334255',\n",
       "     u'X17267',\n",
       "     u'X68248'],\n",
       "    u'protein': [u'AAI62839',\n",
       "     u'AAI62846',\n",
       "     u'AAY67919',\n",
       "     u'CAA35171',\n",
       "     u'CAA48319',\n",
       "     u'CAD59113',\n",
       "     u'NP_571194',\n",
       "     u'P15861'],\n",
       "    u'rna': [u'BC162839', u'BC162846', u'DQ060541', u'NM_131119']},\n",
       "   u'alias': [u'ZF-22', u'ZF22', u'hox-2.2', u'hox-B6', u'hoxb6', u'z-39'],\n",
       "   u'ensembl': {u'gene': u'ENSDARG00000010630',\n",
       "    u'protein': u'ENSDARP00000002263',\n",
       "    u'transcript': u'ENSDART00000024256'},\n",
       "   u'entrezgene': 30341,\n",
       "   u'exons': {u'NM_131119': {u'cdsend': 23574896,\n",
       "     u'cdsstart': 23573155,\n",
       "     u'chr': u'3',\n",
       "     u'exons': [[23573155, 23573585], [23574639, 23574896]],\n",
       "     u'strand': 1,\n",
       "     u'txend': 23574896,\n",
       "     u'txstart': 23573155}},\n",
       "   u'genomic_pos': {u'chr': u'3',\n",
       "    u'end': 23575198,\n",
       "    u'start': 23573156,\n",
       "    u'strand': 1},\n",
       "   u'go': {u'BP': [{u'evidence': u'IEA',\n",
       "      u'id': u'GO:0006351',\n",
       "      u'term': u'transcription, DNA-templated'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0006355',\n",
       "      u'term': u'regulation of transcription, DNA-templated'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0007275',\n",
       "      u'term': u'multicellular organismal development'}],\n",
       "    u'CC': {u'evidence': u'IEA', u'id': u'GO:0005634', u'term': u'nucleus'},\n",
       "    u'MF': [{u'evidence': u'IEA',\n",
       "      u'id': u'GO:0003677',\n",
       "      u'term': u'DNA binding'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0003700',\n",
       "      u'term': u'transcription factor activity, sequence-specific DNA binding'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0043565',\n",
       "      u'term': u'sequence-specific DNA binding'}]},\n",
       "   u'homologene': {u'genes': [[9606, 3216],\n",
       "     [10090, 15414],\n",
       "     [10116, 497986],\n",
       "     [7955, 30341],\n",
       "     [7955, 58053],\n",
       "     [8364, 100124320],\n",
       "     [9031, 777548],\n",
       "     [9598, 749195],\n",
       "     [9615, 491056],\n",
       "     [9913, 100337380]],\n",
       "    u'id': 7366},\n",
       "   u'interpro': [{u'desc': u'Homeobox domain',\n",
       "     u'id': u'IPR001356',\n",
       "     u'short_desc': u'Homeobox_dom'},\n",
       "    {u'desc': u'Homeobox protein, antennapedia type',\n",
       "     u'id': u'IPR017995',\n",
       "     u'short_desc': u'Homeobox_antennapedia'},\n",
       "    {u'desc': u'Homeobox protein, antennapedia type, conserved site',\n",
       "     u'id': u'IPR001827',\n",
       "     u'short_desc': u'Homeobox_Antennapedia_CS'},\n",
       "    {u'desc': u'Homeobox, conserved site',\n",
       "     u'id': u'IPR017970',\n",
       "     u'short_desc': u'Homeobox_CS'},\n",
       "    {u'desc': u'Homeodomain, metazoa',\n",
       "     u'id': u'IPR020479',\n",
       "     u'short_desc': u'Homeobox_metazoa'},\n",
       "    {u'desc': u'Homeodomain-like',\n",
       "     u'id': u'IPR009057',\n",
       "     u'short_desc': u'Homeodomain-like'}],\n",
       "   u'ipi': u'IPI00512770',\n",
       "   u'name': u'homeobox B6a',\n",
       "   u'pfam': u'PF00046',\n",
       "   u'pir': u'S32563',\n",
       "   u'prosite': u'PS50071',\n",
       "   u'refseq': {u'genomic': [u'NC_007114', u'NW_003334255'],\n",
       "    u'protein': u'NP_571194',\n",
       "    u'rna': u'NM_131119'},\n",
       "   u'symbol': u'hoxb6a',\n",
       "   u'taxid': 7955,\n",
       "   u'type_of_gene': u'protein-coding',\n",
       "   u'unigene': u'Dr.132967',\n",
       "   u'uniprot': {u'Swiss-Prot': u'P15861'}},\n",
       "  {u'Vega': u'OTTDARG00000019756',\n",
       "   u'ZFIN': u'ZDB-GENE-980526-70',\n",
       "   u'_id': u'30317',\n",
       "   u'_score': 0.78003955,\n",
       "   u'_timestamp': u'2015-08-02T00:00:00',\n",
       "   u'accession': {u'genomic': [u'AL645782',\n",
       "     u'BX927395',\n",
       "     u'DU644523',\n",
       "     u'NC_007114',\n",
       "     u'NW_003334255',\n",
       "     u'X07381',\n",
       "     u'X68248'],\n",
       "    u'protein': [u'AAH71493',\n",
       "     u'AAY67918',\n",
       "     u'CAA30293',\n",
       "     u'CAA30294',\n",
       "     u'CAA31290',\n",
       "     u'CAA48320',\n",
       "     u'CAD59114',\n",
       "     u'NP_571176',\n",
       "     u'P09014'],\n",
       "    u'rna': [u'BC071493', u'DQ060540', u'NM_131101', u'X12802']},\n",
       "   u'alias': [u'ZF-21',\n",
       "    u'ZF21',\n",
       "    u'hox-2.1',\n",
       "    u'hox-B5',\n",
       "    u'hoxb5',\n",
       "    u'wu:fc38c09',\n",
       "    u'z-18'],\n",
       "   u'ensembl': {u'gene': u'ENSDARG00000013057',\n",
       "    u'protein': u'ENSDARP00000004270',\n",
       "    u'transcript': u'ENSDART00000025449'},\n",
       "   u'entrezgene': 30317,\n",
       "   u'exons': {u'NM_131101': {u'cdsend': 23578681,\n",
       "     u'cdsstart': 23577238,\n",
       "     u'chr': u'3',\n",
       "     u'exons': [[23577151, 23577818], [23578433, 23579546]],\n",
       "     u'strand': 1,\n",
       "     u'txend': 23579546,\n",
       "     u'txstart': 23577151}},\n",
       "   u'generif': [{u'pubmed': u'18832391',\n",
       "     u'text': u'Conserved non-coding elements (CNEs) near the zebrafish hoxb5 genes, were characterized.'}],\n",
       "   u'genomic_pos': {u'chr': u'3',\n",
       "    u'end': 23579547,\n",
       "    u'start': 23577143,\n",
       "    u'strand': 1},\n",
       "   u'go': {u'BP': [{u'evidence': u'IMP',\n",
       "      u'id': u'GO:0001708',\n",
       "      u'pubmed': 11440248,\n",
       "      u'term': u'cell fate specification'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0006351',\n",
       "      u'term': u'transcription, DNA-templated'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0006355',\n",
       "      u'term': u'regulation of transcription, DNA-templated'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0007275',\n",
       "      u'term': u'multicellular organismal development'},\n",
       "     {u'evidence': u'IMP',\n",
       "      u'id': u'GO:0007420',\n",
       "      u'pubmed': 11440248,\n",
       "      u'term': u'brain development'}],\n",
       "    u'CC': {u'evidence': u'IEA', u'id': u'GO:0005634', u'term': u'nucleus'},\n",
       "    u'MF': [{u'evidence': u'IEA',\n",
       "      u'id': u'GO:0003677',\n",
       "      u'term': u'DNA binding'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0003700',\n",
       "      u'term': u'transcription factor activity, sequence-specific DNA binding'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0043565',\n",
       "      u'term': u'sequence-specific DNA binding'}]},\n",
       "   u'homologene': {u'genes': [[7955, 30317], [7955, 58052]], u'id': 137877},\n",
       "   u'interpro': [{u'desc': u'Homeobox domain',\n",
       "     u'id': u'IPR001356',\n",
       "     u'short_desc': u'Homeobox_dom'},\n",
       "    {u'desc': u'Homeobox protein, antennapedia type',\n",
       "     u'id': u'IPR017995',\n",
       "     u'short_desc': u'Homeobox_antennapedia'},\n",
       "    {u'desc': u'Homeobox protein, antennapedia type, conserved site',\n",
       "     u'id': u'IPR001827',\n",
       "     u'short_desc': u'Homeobox_Antennapedia_CS'},\n",
       "    {u'desc': u'Homeobox, conserved site',\n",
       "     u'id': u'IPR017970',\n",
       "     u'short_desc': u'Homeobox_CS'},\n",
       "    {u'desc': u'Homeodomain, metazoa',\n",
       "     u'id': u'IPR020479',\n",
       "     u'short_desc': u'Homeobox_metazoa'},\n",
       "    {u'desc': u'Homeodomain-like',\n",
       "     u'id': u'IPR009057',\n",
       "     u'short_desc': u'Homeodomain-like'}],\n",
       "   u'ipi': u'IPI00493376',\n",
       "   u'name': u'homeobox B5a',\n",
       "   u'pfam': u'PF00046',\n",
       "   u'pir': u'S03671',\n",
       "   u'prosite': u'PS50071',\n",
       "   u'refseq': {u'genomic': [u'NC_007114', u'NW_003334255'],\n",
       "    u'protein': u'NP_571176',\n",
       "    u'rna': u'NM_131101'},\n",
       "   u'retired': 324688,\n",
       "   u'symbol': u'hoxb5a',\n",
       "   u'taxid': 7955,\n",
       "   u'type_of_gene': u'protein-coding',\n",
       "   u'unigene': u'Dr.35792',\n",
       "   u'uniprot': {u'Swiss-Prot': u'P09014'}},\n",
       "  {u'ZFIN': u'ZDB-GENE-990415-118',\n",
       "   u'_id': u'100006598',\n",
       "   u'_score': 0.63036716,\n",
       "   u'_timestamp': u'2015-05-24T00:00:00',\n",
       "   u'accession': {u'genomic': [u'BX322661',\n",
       "     u'NC_007120',\n",
       "     u'NW_003334755',\n",
       "     u'X87753'],\n",
       "    u'protein': [u'AAY87055',\n",
       "     u'CAA61032',\n",
       "     u'CAA74882',\n",
       "     u'NP_001119958',\n",
       "     u'Q90471',\n",
       "     u'XP_005172570'],\n",
       "    u'rna': [u'CB359704',\n",
       "     u'DQ069273',\n",
       "     u'NM_001126486',\n",
       "     u'XM_005172513',\n",
       "     u'Y14547']},\n",
       "   u'alias': [u'HOX-D12', u'hoxd-12', u'hoxd12'],\n",
       "   u'ensembl': {u'gene': u'ENSDARG00000059263',\n",
       "    u'protein': [u'ENSDARP00000076775',\n",
       "     u'ENSDARP00000121699',\n",
       "     u'ENSDARP00000129883'],\n",
       "    u'transcript': [u'ENSDART00000082339',\n",
       "     u'ENSDART00000142842',\n",
       "     u'ENSDART00000165502']},\n",
       "   u'entrezgene': 100006598,\n",
       "   u'exons': {u'NM_001126486': {u'cdsend': 1983783,\n",
       "     u'cdsstart': 1981535,\n",
       "     u'chr': u'9',\n",
       "     u'exons': [[1981084, 1981774], [1983188, 1983807]],\n",
       "     u'strand': -1,\n",
       "     u'txend': 1983807,\n",
       "     u'txstart': 1981084}},\n",
       "   u'genomic_pos': {u'chr': u'9',\n",
       "    u'end': 1985217,\n",
       "    u'start': 1981038,\n",
       "    u'strand': -1},\n",
       "   u'go': {u'BP': [{u'evidence': u'IEA',\n",
       "      u'id': u'GO:0006351',\n",
       "      u'term': u'transcription, DNA-templated'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0006355',\n",
       "      u'term': u'regulation of transcription, DNA-templated'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0007275',\n",
       "      u'term': u'multicellular organismal development'}],\n",
       "    u'CC': {u'evidence': u'IEA', u'id': u'GO:0005634', u'term': u'nucleus'},\n",
       "    u'MF': [{u'evidence': u'IEA',\n",
       "      u'id': u'GO:0003677',\n",
       "      u'term': u'DNA binding'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0043565',\n",
       "      u'term': u'sequence-specific DNA binding'}]},\n",
       "   u'homologene': {u'genes': [[9606, 3238],\n",
       "     [10090, 15432],\n",
       "     [10116, 366082],\n",
       "     [7955, 100006598],\n",
       "     [9031, 396178],\n",
       "     [9544, 100428778],\n",
       "     [9598, 459760],\n",
       "     [9615, 488413],\n",
       "     [9913, 520423]],\n",
       "    u'id': 7369},\n",
       "   u'interpro': [{u'desc': u'Homeobox domain',\n",
       "     u'id': u'IPR001356',\n",
       "     u'short_desc': u'Homeobox_dom'},\n",
       "    {u'desc': u'Homeobox, conserved site',\n",
       "     u'id': u'IPR017970',\n",
       "     u'short_desc': u'Homeobox_CS'},\n",
       "    {u'desc': u'Homeodomain, metazoa',\n",
       "     u'id': u'IPR020479',\n",
       "     u'short_desc': u'Homeobox_metazoa'},\n",
       "    {u'desc': u'Homeodomain-like',\n",
       "     u'id': u'IPR009057',\n",
       "     u'short_desc': u'Homeodomain-like'}],\n",
       "   u'ipi': u'IPI00499135',\n",
       "   u'name': u'homeobox D12a',\n",
       "   u'pfam': u'PF00046',\n",
       "   u'prosite': u'PS50071',\n",
       "   u'refseq': {u'genomic': [u'NC_007120', u'NW_003334755'],\n",
       "    u'protein': [u'NP_001119958', u'XP_005172570'],\n",
       "    u'rna': [u'NM_001126486', u'XM_005172513']},\n",
       "   u'retired': 30406,\n",
       "   u'symbol': u'hoxd12a',\n",
       "   u'taxid': 7955,\n",
       "   u'type_of_gene': u'protein-coding',\n",
       "   u'unigene': [u'Dr.75799', u'Dr.151901'],\n",
       "   u'uniprot': {u'Swiss-Prot': u'Q90471', u'TrEMBL': u'B0S5Y9'}},\n",
       "  {u'Vega': u'OTTDARG00000016368',\n",
       "   u'ZFIN': u'ZDB-GENE-980526-533',\n",
       "   u'_id': u'30379',\n",
       "   u'_score': 0.63036716,\n",
       "   u'_timestamp': u'2015-09-13T00:00:00',\n",
       "   u'accession': {u'genomic': [u'BX005254',\n",
       "     u'M18470',\n",
       "     u'NC_007134',\n",
       "     u'NW_008805325',\n",
       "     u'X68324'],\n",
       "    u'protein': [u'AAA50029',\n",
       "     u'AAI63205',\n",
       "     u'CAA48399',\n",
       "     u'CAA74874',\n",
       "     u'NP_571219',\n",
       "     u'P09074'],\n",
       "    u'rna': [u'BC163205', u'NM_131144', u'Y14539']},\n",
       "   u'alias': [u'ZF-25', u'hox-3.4', u'hoxc5', u'z-142'],\n",
       "   u'ensembl': {u'gene': u'ENSDARG00000070340',\n",
       "    u'protein': [u'ENSDARP00000093908', u'ENSDARP00000121862'],\n",
       "    u'transcript': [u'ENSDART00000103134', u'ENSDART00000139319']},\n",
       "   u'entrezgene': 30379,\n",
       "   u'exons': {u'NM_131144': {u'cdsend': 36021454,\n",
       "     u'cdsstart': 36019934,\n",
       "     u'chr': u'23',\n",
       "     u'exons': [[36019870, 36020421], [36021239, 36021708]],\n",
       "     u'strand': 1,\n",
       "     u'txend': 36021708,\n",
       "     u'txstart': 36019870}},\n",
       "   u'genomic_pos': {u'chr': u'23',\n",
       "    u'end': 36021708,\n",
       "    u'start': 36019935,\n",
       "    u'strand': 1},\n",
       "   u'go': {u'BP': [{u'evidence': u'IEA',\n",
       "      u'id': u'GO:0006351',\n",
       "      u'term': u'transcription, DNA-templated'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0006355',\n",
       "      u'term': u'regulation of transcription, DNA-templated'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0007275',\n",
       "      u'term': u'multicellular organismal development'}],\n",
       "    u'CC': {u'evidence': u'IEA', u'id': u'GO:0005634', u'term': u'nucleus'},\n",
       "    u'MF': [{u'evidence': u'IEA',\n",
       "      u'id': u'GO:0003677',\n",
       "      u'term': u'DNA binding'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0003700',\n",
       "      u'term': u'transcription factor activity, sequence-specific DNA binding'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0043565',\n",
       "      u'term': u'sequence-specific DNA binding'}]},\n",
       "   u'homologene': {u'genes': [[9606, 3222],\n",
       "     [10090, 15424],\n",
       "     [10116, 315341],\n",
       "     [7955, 30379],\n",
       "     [8364, 100487879],\n",
       "     [9598, 467010],\n",
       "     [9615, 102152615],\n",
       "     [9913, 100140629]],\n",
       "    u'id': 41296},\n",
       "   u'interpro': [{u'desc': u'Homeobox domain',\n",
       "     u'id': u'IPR001356',\n",
       "     u'short_desc': u'Homeobox_dom'},\n",
       "    {u'desc': u'Homeobox protein, antennapedia type',\n",
       "     u'id': u'IPR017995',\n",
       "     u'short_desc': u'Homeobox_antennapedia'},\n",
       "    {u'desc': u'Homeobox protein, antennapedia type, conserved site',\n",
       "     u'id': u'IPR001827',\n",
       "     u'short_desc': u'Homeobox_Antennapedia_CS'},\n",
       "    {u'desc': u'Homeobox, conserved site',\n",
       "     u'id': u'IPR017970',\n",
       "     u'short_desc': u'Homeobox_CS'},\n",
       "    {u'desc': u'Homeodomain, metazoa',\n",
       "     u'id': u'IPR020479',\n",
       "     u'short_desc': u'Homeobox_metazoa'},\n",
       "    {u'desc': u'Homeodomain-like',\n",
       "     u'id': u'IPR009057',\n",
       "     u'short_desc': u'Homeodomain-like'}],\n",
       "   u'ipi': u'IPI00934138',\n",
       "   u'name': u'homeobox C5a',\n",
       "   u'pfam': u'PF00046',\n",
       "   u'pir': u'S48125',\n",
       "   u'prosite': u'PS50071',\n",
       "   u'refseq': {u'genomic': [u'NC_007134', u'NW_008805325'],\n",
       "    u'protein': u'NP_571219',\n",
       "    u'rna': u'NM_131144'},\n",
       "   u'symbol': u'hoxc5a',\n",
       "   u'taxid': 7955,\n",
       "   u'type_of_gene': u'protein-coding',\n",
       "   u'unigene': u'Dr.132198',\n",
       "   u'uniprot': {u'Swiss-Prot': u'P09074'}},\n",
       "  {u'Vega': u'OTTDARG00000017442',\n",
       "   u'ZFIN': u'ZDB-GENE-990415-116',\n",
       "   u'_id': u'30404',\n",
       "   u'_score': 0.5724417,\n",
       "   u'_timestamp': u'2015-05-24T00:00:00',\n",
       "   u'accession': {u'genomic': [u'BX322661',\n",
       "     u'NC_007120',\n",
       "     u'NW_003334755',\n",
       "     u'U51403',\n",
       "     u'U51421',\n",
       "     u'X87750'],\n",
       "    u'protein': [u'AAB38626',\n",
       "     u'AAB38643',\n",
       "     u'AAH71329',\n",
       "     u'AAI53659',\n",
       "     u'CAA61029',\n",
       "     u'CAA74877',\n",
       "     u'NP_571241',\n",
       "     u'Q90469'],\n",
       "    u'rna': [u'BC071329',\n",
       "     u'BC153658',\n",
       "     u'CK864624',\n",
       "     u'CT704711',\n",
       "     u'NM_131166',\n",
       "     u'Y14542']},\n",
       "   u'alias': [u'HOX-D10', u'hoxd-10', u'hoxd10', u'z-82'],\n",
       "   u'ensembl': {u'gene': u'ENSDARG00000057859',\n",
       "    u'protein': u'ENSDARP00000075055',\n",
       "    u'transcript': u'ENSDART00000080608'},\n",
       "   u'entrezgene': 30404,\n",
       "   u'exons': {u'NM_131166': {u'cdsend': 1969136,\n",
       "     u'cdsstart': 1967422,\n",
       "     u'chr': u'9',\n",
       "     u'exons': [[1966908, 1967697], [1968412, 1969250]],\n",
       "     u'strand': -1,\n",
       "     u'txend': 1969250,\n",
       "     u'txstart': 1966908}},\n",
       "   u'genomic_pos': {u'chr': u'9',\n",
       "    u'end': 1969274,\n",
       "    u'start': 1966910,\n",
       "    u'strand': -1},\n",
       "   u'go': {u'BP': [{u'evidence': u'IEA',\n",
       "      u'id': u'GO:0006351',\n",
       "      u'term': u'transcription, DNA-templated'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0006355',\n",
       "      u'term': u'regulation of transcription, DNA-templated'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0007275',\n",
       "      u'term': u'multicellular organismal development'}],\n",
       "    u'CC': {u'evidence': u'IEA', u'id': u'GO:0005634', u'term': u'nucleus'},\n",
       "    u'MF': [{u'evidence': u'IEA',\n",
       "      u'id': u'GO:0003677',\n",
       "      u'term': u'DNA binding'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0043565',\n",
       "      u'term': u'sequence-specific DNA binding'}]},\n",
       "   u'homologene': {u'genes': [[9606, 3236],\n",
       "     [10090, 15430],\n",
       "     [10116, 303991],\n",
       "     [7955, 30404],\n",
       "     [8364, 100038085],\n",
       "     [9031, 771246],\n",
       "     [9598, 739607],\n",
       "     [9615, 609973],\n",
       "     [9913, 540251]],\n",
       "    u'id': 1619},\n",
       "   u'interpro': [{u'desc': u'Homeobox domain',\n",
       "     u'id': u'IPR001356',\n",
       "     u'short_desc': u'Homeobox_dom'},\n",
       "    {u'desc': u'Homeobox, conserved site',\n",
       "     u'id': u'IPR017970',\n",
       "     u'short_desc': u'Homeobox_CS'},\n",
       "    {u'desc': u'Homeodomain, metazoa',\n",
       "     u'id': u'IPR020479',\n",
       "     u'short_desc': u'Homeobox_metazoa'},\n",
       "    {u'desc': u'Homeodomain-like',\n",
       "     u'id': u'IPR009057',\n",
       "     u'short_desc': u'Homeodomain-like'}],\n",
       "   u'ipi': [u'IPI00551911', u'IPI00996139'],\n",
       "   u'name': u'homeobox D10a',\n",
       "   u'pfam': u'PF00046',\n",
       "   u'prosite': u'PS50071',\n",
       "   u'refseq': {u'genomic': [u'NC_007120', u'NW_003334755'],\n",
       "    u'protein': u'NP_571241',\n",
       "    u'rna': u'NM_131166'},\n",
       "   u'symbol': u'hoxd10a',\n",
       "   u'taxid': 7955,\n",
       "   u'type_of_gene': u'protein-coding',\n",
       "   u'unigene': u'Dr.75795',\n",
       "   u'uniprot': {u'Swiss-Prot': u'Q90469', u'TrEMBL': u'E9QBR9'}},\n",
       "  {u'Vega': u'OTTDARG00000016363',\n",
       "   u'ZFIN': u'ZDB-GENE-990415-113',\n",
       "   u'_id': u'30346',\n",
       "   u'_score': 0.5724417,\n",
       "   u'_timestamp': u'2015-08-02T00:00:00',\n",
       "   u'accession': {u'genomic': [u'AF071265',\n",
       "     u'BX005254',\n",
       "     u'DU644831',\n",
       "     u'NC_007134',\n",
       "     u'NW_008805325',\n",
       "     u'X17266'],\n",
       "    u'protein': [u'AAD15958',\n",
       "     u'AAH83307',\n",
       "     u'AAI65049',\n",
       "     u'AAY67932',\n",
       "     u'CAA35170',\n",
       "     u'CAA74881',\n",
       "     u'NP_571198',\n",
       "     u'P15862'],\n",
       "    u'rna': [u'BC083307', u'BC165049', u'DQ060554', u'NM_131123', u'Y14546']},\n",
       "   u'alias': [u'HOX-C6', u'ZF-61', u'hoxc-6', u'hoxc6', u'zgc:101882'],\n",
       "   u'ensembl': {u'gene': u'ENSDARG00000070343',\n",
       "    u'protein': [u'ENSDARP00000093809', u'ENSDARP00000107797'],\n",
       "    u'transcript': [u'ENSDART00000103035', u'ENSDART00000130090']},\n",
       "   u'entrezgene': 30346,\n",
       "   u'exons': {u'NM_131123': {u'cdsend': 36017974,\n",
       "     u'cdsstart': 36016668,\n",
       "     u'chr': u'23',\n",
       "     u'exons': [[36016668, 36017071], [36017681, 36017974]],\n",
       "     u'strand': 1,\n",
       "     u'txend': 36017974,\n",
       "     u'txstart': 36016668}},\n",
       "   u'genomic_pos': {u'chr': u'23',\n",
       "    u'end': 36018761,\n",
       "    u'start': 36016455,\n",
       "    u'strand': 1},\n",
       "   u'go': {u'BP': [{u'evidence': u'IEA',\n",
       "      u'id': u'GO:0006351',\n",
       "      u'term': u'transcription, DNA-templated'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0006355',\n",
       "      u'term': u'regulation of transcription, DNA-templated'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0007275',\n",
       "      u'term': u'multicellular organismal development'}],\n",
       "    u'CC': {u'evidence': u'IEA', u'id': u'GO:0005634', u'term': u'nucleus'},\n",
       "    u'MF': [{u'evidence': u'IEA',\n",
       "      u'id': u'GO:0003677',\n",
       "      u'term': u'DNA binding'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0003700',\n",
       "      u'term': u'transcription factor activity, sequence-specific DNA binding'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0043565',\n",
       "      u'term': u'sequence-specific DNA binding'}]},\n",
       "   u'homologene': {u'genes': [[9606, 3223],\n",
       "     [10090, 15425],\n",
       "     [10116, 252885],\n",
       "     [7955, 30346],\n",
       "     [8364, 101733663],\n",
       "     [9031, 100858368],\n",
       "     [9598, 100607977],\n",
       "     [9615, 486499],\n",
       "     [9913, 317660]],\n",
       "    u'id': 19533},\n",
       "   u'interpro': [{u'desc': u'Homeobox domain',\n",
       "     u'id': u'IPR001356',\n",
       "     u'short_desc': u'Homeobox_dom'},\n",
       "    {u'desc': u'Homeobox protein, antennapedia type, conserved site',\n",
       "     u'id': u'IPR001827',\n",
       "     u'short_desc': u'Homeobox_Antennapedia_CS'},\n",
       "    {u'desc': u'Homeobox, conserved site',\n",
       "     u'id': u'IPR017970',\n",
       "     u'short_desc': u'Homeobox_CS'},\n",
       "    {u'desc': u'Homeodomain, metazoa',\n",
       "     u'id': u'IPR020479',\n",
       "     u'short_desc': u'Homeobox_metazoa'},\n",
       "    {u'desc': u'Homeodomain-like',\n",
       "     u'id': u'IPR009057',\n",
       "     u'short_desc': u'Homeodomain-like'}],\n",
       "   u'ipi': u'IPI00494357',\n",
       "   u'name': u'homeobox C6a',\n",
       "   u'pfam': u'PF00046',\n",
       "   u'pir': u'S08639',\n",
       "   u'prosite': u'PS50071',\n",
       "   u'refseq': {u'genomic': [u'NC_007134', u'NW_008805325'],\n",
       "    u'protein': u'NP_571198',\n",
       "    u'rna': u'NM_131123'},\n",
       "   u'symbol': u'hoxc6a',\n",
       "   u'taxid': 7955,\n",
       "   u'type_of_gene': u'protein-coding',\n",
       "   u'unigene': u'Dr.132203',\n",
       "   u'uniprot': {u'Swiss-Prot': u'P15862'}},\n",
       "  {u'ZFIN': u'ZDB-GENE-980526-532',\n",
       "   u'_id': u'30421',\n",
       "   u'_score': 0.53390026,\n",
       "   u'_timestamp': u'2015-05-24T00:00:00',\n",
       "   u'accession': {u'genomic': [u'AB175067',\n",
       "     u'CU013512',\n",
       "     u'NC_007134',\n",
       "     u'NW_008805325'],\n",
       "    u'protein': [u'AAH95299',\n",
       "     u'AAI65158',\n",
       "     u'AAY67929',\n",
       "     u'CAA42691',\n",
       "     u'NP_001128157',\n",
       "     u'P28174',\n",
       "     u'XP_005162422',\n",
       "     u'XP_005162423',\n",
       "     u'XP_005162424'],\n",
       "    u'rna': [u'AB175333',\n",
       "     u'BC095299',\n",
       "     u'BC154496',\n",
       "     u'BC165158',\n",
       "     u'DQ060551',\n",
       "     u'NM_001134685',\n",
       "     u'X60095',\n",
       "     u'XM_005162365',\n",
       "     u'XM_005162366',\n",
       "     u'XM_005162367']},\n",
       "   u'alias': [u'HOX-114', u'zgc:110503'],\n",
       "   u'ensembl': {u'gene': u'ENSDARG00000070339',\n",
       "    u'protein': [u'ENSDARP00000076907',\n",
       "     u'ENSDARP00000076908',\n",
       "     u'ENSDARP00000129026',\n",
       "     u'ENSDARP00000129398'],\n",
       "    u'transcript': [u'ENSDART00000082472',\n",
       "     u'ENSDART00000082473',\n",
       "     u'ENSDART00000128533',\n",
       "     u'ENSDART00000137507',\n",
       "     u'ENSDART00000154825']},\n",
       "   u'entrezgene': 30421,\n",
       "   u'exons': {u'NM_001134685': {u'cdsend': 36049265,\n",
       "     u'cdsstart': 36046234,\n",
       "     u'chr': u'23',\n",
       "     u'exons': [[35988493, 35988604],\n",
       "      [36028419, 36028471],\n",
       "      [36046220, 36046390],\n",
       "      [36046580, 36046848],\n",
       "      [36048897, 36050239]],\n",
       "     u'strand': 1,\n",
       "     u'txend': 36050239,\n",
       "     u'txstart': 35988493}},\n",
       "   u'genomic_pos': {u'chr': u'23',\n",
       "    u'end': 36050568,\n",
       "    u'start': 35988416,\n",
       "    u'strand': 1},\n",
       "   u'go': {u'BP': [{u'evidence': u'IEA',\n",
       "      u'id': u'GO:0006355',\n",
       "      u'term': u'regulation of transcription, DNA-templated'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0007275',\n",
       "      u'term': u'multicellular organismal development'}],\n",
       "    u'CC': {u'evidence': u'IEA', u'id': u'GO:0005634', u'term': u'nucleus'},\n",
       "    u'MF': [{u'evidence': u'IEA',\n",
       "      u'id': u'GO:0003677',\n",
       "      u'term': u'DNA binding'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0043565',\n",
       "      u'term': u'sequence-specific DNA binding'}]},\n",
       "   u'interpro': [{u'desc': u'Homeobox domain',\n",
       "     u'id': u'IPR001356',\n",
       "     u'short_desc': u'Homeobox_dom'},\n",
       "    {u'desc': u'Homeobox, conserved site',\n",
       "     u'id': u'IPR017970',\n",
       "     u'short_desc': u'Homeobox_CS'},\n",
       "    {u'desc': u'Homeodomain, metazoa',\n",
       "     u'id': u'IPR020479',\n",
       "     u'short_desc': u'Homeobox_metazoa'},\n",
       "    {u'desc': u'Homeodomain-like',\n",
       "     u'id': u'IPR009057',\n",
       "     u'short_desc': u'Homeodomain-like'}],\n",
       "   u'ipi': u'IPI00931602',\n",
       "   u'name': u'homeo box C3a',\n",
       "   u'pfam': u'PF00046',\n",
       "   u'prosite': u'PS50071',\n",
       "   u'refseq': {u'genomic': [u'NC_007134', u'NW_008805325'],\n",
       "    u'protein': [u'NP_001128157',\n",
       "     u'XP_005162422',\n",
       "     u'XP_005162423',\n",
       "     u'XP_005162424'],\n",
       "    u'rna': [u'NM_001134685',\n",
       "     u'XM_005162365',\n",
       "     u'XM_005162366',\n",
       "     u'XM_005162367']},\n",
       "   u'retired': 100151318,\n",
       "   u'symbol': u'hoxc3a',\n",
       "   u'taxid': 7955,\n",
       "   u'type_of_gene': u'protein-coding',\n",
       "   u'unigene': [u'Dr.510', u'Dr.132202']},\n",
       "  {u'Vega': u'OTTDARG00000015149',\n",
       "   u'ZFIN': u'ZDB-GENE-000823-5',\n",
       "   u'_id': u'58051',\n",
       "   u'_score': 0.5175482,\n",
       "   u'_timestamp': u'2015-05-31T00:00:00',\n",
       "   u'accession': {u'genomic': [u'AF071243',\n",
       "     u'AL645756',\n",
       "     u'CR382300',\n",
       "     u'DU644665',\n",
       "     u'NC_007130',\n",
       "     u'NW_001877953'],\n",
       "    u'protein': [u'AAD15937',\n",
       "     u'AAI62701',\n",
       "     u'AAY67909',\n",
       "     u'CAC34565',\n",
       "     u'CAC34566',\n",
       "     u'CAD52136',\n",
       "     u'CAD52137',\n",
       "     u'NP_571611',\n",
       "     u'Q98SI1',\n",
       "     u'XP_005159579',\n",
       "     u'XP_009292446',\n",
       "     u'XP_009292447'],\n",
       "    u'rna': [u'AJ306430',\n",
       "     u'AJ306431',\n",
       "     u'BC162701',\n",
       "     u'DQ060531',\n",
       "     u'NM_131536',\n",
       "     u'XM_005159522',\n",
       "     u'XM_009294171',\n",
       "     u'XM_009294172']},\n",
       "   u'alias': [u'Hox-a1', u'hoxa3a', u'id:ibd3526', u'z-4'],\n",
       "   u'ensembl': {u'gene': u'ENSDARG00000104307',\n",
       "    u'protein': [u'ENSDARP00000133272', u'ENSDARP00000133812'],\n",
       "    u'transcript': [u'ENSDART00000163546', u'ENSDART00000167757']},\n",
       "   u'entrezgene': 58051,\n",
       "   u'exons': {u'NM_131536': {u'cdsend': 20214446,\n",
       "     u'cdsstart': 20213261,\n",
       "     u'chr': u'19',\n",
       "     u'exons': [[20213165, 20213901], [20214096, 20214468]],\n",
       "     u'strand': 1,\n",
       "     u'txend': 20214468,\n",
       "     u'txstart': 20213165}},\n",
       "   u'genomic_pos': {u'chr': u'19',\n",
       "    u'end': 20214801,\n",
       "    u'start': 20213166,\n",
       "    u'strand': 1},\n",
       "   u'go': {u'BP': [{u'evidence': u'IEA',\n",
       "      u'id': u'GO:0006351',\n",
       "      u'term': u'transcription, DNA-templated'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0006355',\n",
       "      u'term': u'regulation of transcription, DNA-templated'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0007275',\n",
       "      u'term': u'multicellular organismal development'}],\n",
       "    u'CC': {u'evidence': u'IEA', u'id': u'GO:0005634', u'term': u'nucleus'},\n",
       "    u'MF': [{u'evidence': u'IEA',\n",
       "      u'id': u'GO:0003677',\n",
       "      u'term': u'DNA binding'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0043565',\n",
       "      u'term': u'sequence-specific DNA binding'}]},\n",
       "   u'homologene': {u'genes': [[9606, 3198],\n",
       "     [10090, 15394],\n",
       "     [10116, 103690132],\n",
       "     [7955, 58051],\n",
       "     [8364, 493379],\n",
       "     [9031, 374051],\n",
       "     [9544, 700723],\n",
       "     [9598, 463308],\n",
       "     [9615, 482367],\n",
       "     [9913, 537889]],\n",
       "    u'id': 4032},\n",
       "   u'interpro': [{u'desc': u'Homeobox domain',\n",
       "     u'id': u'IPR001356',\n",
       "     u'short_desc': u'Homeobox_dom'},\n",
       "    {u'desc': u'Homeobox, conserved site',\n",
       "     u'id': u'IPR017970',\n",
       "     u'short_desc': u'Homeobox_CS'},\n",
       "    {u'desc': u'Homeodomain, metazoa',\n",
       "     u'id': u'IPR020479',\n",
       "     u'short_desc': u'Homeobox_metazoa'},\n",
       "    {u'desc': u'Homeodomain-like',\n",
       "     u'id': u'IPR009057',\n",
       "     u'short_desc': u'Homeodomain-like'}],\n",
       "   u'ipi': [u'IPI00494329', u'IPI00494805'],\n",
       "   u'name': u'homeobox A1a',\n",
       "   u'pfam': u'PF00046',\n",
       "   u'prosite': u'PS50071',\n",
       "   u'refseq': {u'genomic': [u'NC_007130', u'NW_001877953'],\n",
       "    u'protein': [u'NP_571611',\n",
       "     u'XP_005159579',\n",
       "     u'XP_009292446',\n",
       "     u'XP_009292447'],\n",
       "    u'rna': [u'NM_131536', u'XM_005159522', u'XM_009294171', u'XM_009294172']},\n",
       "   u'retired': 338133,\n",
       "   u'symbol': u'hoxa1a',\n",
       "   u'taxid': 7955,\n",
       "   u'type_of_gene': u'protein-coding',\n",
       "   u'unigene': u'Dr.83046',\n",
       "   u'uniprot': {u'Swiss-Prot': u'Q98SI1', u'TrEMBL': u'B3DHB1'}},\n",
       "  {u'Vega': u'OTTDARG00000019821',\n",
       "   u'ZFIN': u'ZDB-GENE-990415-105',\n",
       "   u'_id': u'30340',\n",
       "   u'_score': 0.5175482,\n",
       "   u'_timestamp': u'2015-08-02T00:00:00',\n",
       "   u'accession': {u'genomic': [u'AF071252',\n",
       "     u'AL645782',\n",
       "     u'BX927395',\n",
       "     u'M24085',\n",
       "     u'NC_007114',\n",
       "     u'NW_003334255'],\n",
       "    u'protein': [u'AAA56866',\n",
       "     u'AAD15945',\n",
       "     u'AAI62840',\n",
       "     u'AAI62844',\n",
       "     u'CAA74284',\n",
       "     u'CAD59115',\n",
       "     u'DAA05783',\n",
       "     u'NP_571193',\n",
       "     u'P22574'],\n",
       "    u'rna': [u'BC162840', u'BC162844', u'BK005804', u'NM_131118', u'Y13946']},\n",
       "   u'alias': [u'HOX-B4', u'Z-17', u'ZF-13', u'hoxb4', u'zf13'],\n",
       "   u'ensembl': {u'gene': u'ENSDARG00000013533',\n",
       "    u'protein': [u'ENSDARP00000008458', u'ENSDARP00000125694'],\n",
       "    u'transcript': [u'ENSDART00000012470', u'ENSDART00000151584']},\n",
       "   u'entrezgene': 30340,\n",
       "   u'exons': {u'NM_131118': {u'cdsend': 23592940,\n",
       "     u'cdsstart': 23591259,\n",
       "     u'chr': u'3',\n",
       "     u'exons': [[23591259, 23591683], [23592623, 23592940]],\n",
       "     u'strand': 1,\n",
       "     u'txend': 23592940,\n",
       "     u'txstart': 23591259}},\n",
       "   u'generif': [{u'pubmed': u'16860306',\n",
       "     u'text': u'the similarity of hoxb3a/Hoxa3 regulatory mechanisms reflect the shared descent of both genes from a single ancestral paralog, expression of both genes in the posterior hindbrain and spinal cord in embryo'},\n",
       "    {u'pubmed': u'18627100',\n",
       "     u'text': u'hoxb4a in combination with its paralogs may play a significant role in the development of precerebellar, vagal, and pectoral fin neuronal subtypes'},\n",
       "    {u'pubmed': u'19536294',\n",
       "     u'text': u'continuous presence of hoxb4a positive neurons may provide a developmental plasticity for behavior-specific circuits to accommodate experience- and growth-related changes'}],\n",
       "   u'genomic_pos': {u'chr': u'3',\n",
       "    u'end': 23593434,\n",
       "    u'start': 23580291,\n",
       "    u'strand': 1},\n",
       "   u'go': {u'BP': [{u'evidence': u'IEA',\n",
       "      u'id': u'GO:0006351',\n",
       "      u'term': u'transcription, DNA-templated'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0006355',\n",
       "      u'term': u'regulation of transcription, DNA-templated'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0007275',\n",
       "      u'term': u'multicellular organismal development'}],\n",
       "    u'CC': {u'evidence': u'IEA', u'id': u'GO:0005634', u'term': u'nucleus'},\n",
       "    u'MF': [{u'evidence': u'IEA',\n",
       "      u'id': u'GO:0003677',\n",
       "      u'term': u'DNA binding'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0003700',\n",
       "      u'term': u'transcription factor activity, sequence-specific DNA binding'},\n",
       "     {u'evidence': u'IEA',\n",
       "      u'id': u'GO:0043565',\n",
       "      u'term': u'sequence-specific DNA binding'}]},\n",
       "   u'homologene': {u'genes': [[9606, 3214],\n",
       "     [10090, 15412],\n",
       "     [10116, 497988],\n",
       "     [7955, 30340],\n",
       "     [8364, 548393],\n",
       "     [9031, 396230],\n",
       "     [9598, 749218],\n",
       "     [9615, 608915],\n",
       "     [9913, 768240]],\n",
       "    u'id': 32095},\n",
       "   u'interpro': [{u'desc': u'Homeobox domain',\n",
       "     u'id': u'IPR001356',\n",
       "     u'short_desc': u'Homeobox_dom'},\n",
       "    {u'desc': u'Homeobox protein, antennapedia type',\n",
       "     u'id': u'IPR017995',\n",
       "     u'short_desc': u'Homeobox_antennapedia'},\n",
       "    {u'desc': u'Homeobox protein, antennapedia type, conserved site',\n",
       "     u'id': u'IPR001827',\n",
       "     u'short_desc': u'Homeobox_Antennapedia_CS'},\n",
       "    {u'desc': u'Homeobox, conserved site',\n",
       "     u'id': u'IPR017970',\n",
       "     u'short_desc': u'Homeobox_CS'},\n",
       "    {u'desc': u'Homeodomain, metazoa',\n",
       "     u'id': u'IPR020479',\n",
       "     u'short_desc': u'Homeobox_metazoa'},\n",
       "    {u'desc': u'Homeodomain-like',\n",
       "     u'id': u'IPR009057',\n",
       "     u'short_desc': u'Homeodomain-like'}],\n",
       "   u'ipi': u'IPI00499387',\n",
       "   u'name': u'homeobox B4a',\n",
       "   u'pfam': u'PF00046',\n",
       "   u'pir': u'JT0489',\n",
       "   u'prosite': u'PS50071',\n",
       "   u'refseq': {u'genomic': [u'NC_007114', u'NW_003334255'],\n",
       "    u'protein': u'NP_571193',\n",
       "    u'rna': u'NM_131118'},\n",
       "   u'symbol': u'hoxb4a',\n",
       "   u'taxid': 7955,\n",
       "   u'type_of_gene': u'protein-coding',\n",
       "   u'unigene': u'Dr.133609',\n",
       "   u'uniprot': {u'Swiss-Prot': u'P22574', u'TrEMBL': [u'B3DHP6', u'Q0D251']}}],\n",
       " u'max_score': 0.8095548,\n",
       " u'took': 37,\n",
       " u'total': 9}"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import mygene\n",
    "mg = mygene.MyGeneInfo()\n",
    "\n",
    "mg.query('HOX', fields='all', species='zebrafish', verbose=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "* Looking at individual projects as iterative processes requires 1) and 2).  pweave connects the processing with the description and let's me crank the wheel faster.\n",
    "    - https://github.com/dib-lab/2015-petMarSB/blob/master/doc/2015-petMarSB.draft.texw\n",
    "    - https://github.com/dib-lab/2015-petMarSB/blob/master/doc/2015-petMarSB.draft.tex\n",
    "    - https://github.com/dib-lab/2015-petMarSB/blob/master/doc/2015-petMarSB.draft.pdf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "source": [
    "Putting it all together: we can construct complex analyses from these simple building blocks. This has been put to practice for the [lamprey transcriptome](http://nbviewer.ipython.org/github/dib-lab/2015-petMarSB/blob/master/notebooks/petmar-genome-completeness.ipynb)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "slideshow": {
     "slide_type": "fragment"
    }
   },
   "source": [
    "![proposed craniate phylogenies](http://icb.oxfordjournals.org/content/50/1/130/F2.medium.gif)\n"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Slideshow",
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
